--- a/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd	2015-05-03 22:06:17.232607123 +0300
+++ b/arch/arm/boot/dts/amlogic/meson8b_m201_1G.dtd	2015-05-03 22:13:24.471606866 +0300
@@ -131,12 +131,12 @@
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-		reserve-memory = <0x00800000 0x00100000>;
+		reserve-memory = <0x01800000 0x00100000>;
 		reserve-iomap = "true";
 	        vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
 		4k2k_fb = <0>;
-	        display_size_default = <1280 720 1280 1440 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
+	        display_size_default = <1920 1080 1920 3840 32>;  // osd0:8M, osd1:1m  1280*720*4*3 = 11,059,200
 	};
 
 //$$ DEVICE="deinterlace"
