diff -Naur systemd-231/src/systemctl/systemctl.c systemd-231.patch/src/systemctl/systemctl.c
--- systemd-231/src/systemctl/systemctl.c	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231.patch/src/systemctl/systemctl.c	2016-08-01 10:38:09.126685204 +0200
@@ -6513,6 +6513,11 @@
         return r;
 }
 
+static int nope(sd_bus *bus, char **args) {
+        printf("nope. don't do that\n");
+        return 0;
+}
+
 static void systemctl_help(void) {
 
         pager_open(arg_no_pager, false);
@@ -7730,22 +7735,22 @@
                 { "enable",                2,        VERB_ANY, 0,             enable_unit          },
                 { "disable",               2,        VERB_ANY, 0,             enable_unit          },
                 { "is-enabled",            2,        VERB_ANY, 0,             unit_is_enabled      },
-                { "reenable",              2,        VERB_ANY, 0,             enable_unit          },
-                { "preset",                2,        VERB_ANY, 0,             enable_unit          },
-                { "preset-all",            VERB_ANY, 1,        0,             preset_all           },
-                { "mask",                  2,        VERB_ANY, 0,             enable_unit          },
-                { "unmask",                2,        VERB_ANY, 0,             enable_unit          },
-                { "link",                  2,        VERB_ANY, 0,             enable_unit          },
+                { "reenable",              2,        VERB_ANY, 0,             nope              },
+                { "preset",                2,        VERB_ANY, 0,             nope              },
+                { "preset-all",            VERB_ANY, 1,        0,             nope              },
+                { "mask",                  2,        VERB_ANY, 0,             nope              },
+                { "unmask",                2,        VERB_ANY, 0,             nope              },
+                { "link",                  2,        VERB_ANY, 0,             nope              },
                 { "revert",                2,        VERB_ANY, 0,             enable_unit          },
                 { "switch-root",           2,        VERB_ANY, VERB_NOCHROOT, switch_root          },
                 { "list-dependencies",     VERB_ANY, 2,        VERB_NOCHROOT, list_dependencies    },
-                { "set-default",           2,        2,        0,             set_default          },
+                { "set-default",           2,        2,        0,             nope              },
                 { "get-default",           VERB_ANY, 1,        0,             get_default          },
                 { "set-property",          3,        VERB_ANY, VERB_NOCHROOT, set_property         },
                 { "is-system-running",     VERB_ANY, 1,        0,             is_system_running    },
-                { "add-wants",             3,        VERB_ANY, 0,             add_dependency       },
-                { "add-requires",          3,        VERB_ANY, 0,             add_dependency       },
-                { "edit",                  2,        VERB_ANY, VERB_NOCHROOT, edit                 },
+                { "add-wants",             3,        VERB_ANY, 0,             nope              },
+                { "add-requires",          3,        VERB_ANY, 0,             nope              },
+                { "edit",                  2,        VERB_ANY, 0,             nope              },
                 {}
         };
 
