--- a/drivers/amlogic/wifi/broadcom/broadcm_4354/Makefile~	2015-08-14 08:43:36.000000000 +0200
+++ b/drivers/amlogic/wifi/broadcom/broadcm_4354/Makefile	2015-08-16 10:24:25.044138572 +0200
@@ -14,7 +14,7 @@ DHDCFLAGS = -Wall -Wstrict-prototypes -D
 	-DMIRACAST_AMPDU_SIZE=8                                               \
 	-DSDTEST -DBDC -DDHD_BCMEVENTS -DPROP_TXSTATUS -DPROP_TXSTATUS_VSDB   \
 	-DWL_SUPPORT_BACKPORTED_KPATCHES -DDHDTCPACK_SUPPRESS                 \
-	-Idrivers/amlogic/wifi/broadcom/bcmdhd_4354/ -Idrivers/amlogic/wifi/broadcom/bcmdhd_4354/include
+	-Idrivers/amlogic/wifi/broadcom/broadcm_4354/ -Idrivers/amlogic/wifi/broadcom/broadcm_4354/include
        
 
 DHDCFLAGS += \
