diff --git a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
index 79fc57b..af0307c 100755
--- a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
+++ b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
@@ -114,6 +114,7 @@ const static char *nv_name[] = {
 	"ap6330_nvram.txt",
 	"ap6181_nvram.txt",
 	"ap6335e_nvram.txt",
+	"ap6234_nvram.txt",
 };
 
 #define htod32(i) i
@@ -371,7 +372,7 @@ dhd_conf_set_nv_name_by_chip(dhd_pub_t *dhd, char *dst, char *src)
 					strcpy(&dst[i+1], nv_name[0]);
 						break;	
 		case BCM43340_CHIP_ID: //bcm43341b0ag BCM43340B0
-					strcpy(&dst[i+1], nv_name[0]);
+					strcpy(&dst[i+1], nv_name[4]);
 						break;
 //		case BCM43341_CHIP_ID: //bcm43341b0ag BCM43341B0 
 //					strcpy(&dst[i+1], nv_name[0]);
