diff --git a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
index 79fc57b..3e61dd3 100755
--- a/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
+++ b/drivers/amlogic/wifi/broadcom/broadcm_40181/dhd_config.c
@@ -114,6 +114,8 @@ const static char *nv_name[] = {
 	"ap6330_nvram.txt",
 	"ap6181_nvram.txt",
 	"ap6335e_nvram.txt",
+	"ap6234_nvram.txt",
+	"ap62x2_nvram.txt",
 };
 
 #define htod32(i) i
@@ -371,14 +373,17 @@ dhd_conf_set_nv_name_by_chip(dhd_pub_t *dhd, char *dst, char *src)
 					strcpy(&dst[i+1], nv_name[0]);
 						break;	
 		case BCM43340_CHIP_ID: //bcm43341b0ag BCM43340B0
-					strcpy(&dst[i+1], nv_name[0]);
+			if (chiprev == BCM43340B0_CHIP_REV)
+				strcpy(&dst[i+1], nv_name[4]);
+			else
+				strcpy(&dst[i+1], nv_name[0]);
+						break;
+		case BCM43341_CHIP_ID: //bcm43341b0ag BCM43341B0 
+					strcpy(&dst[i+1], nv_name[4]);
+						break;
+		case BCM4324_CHIP_ID: //bcm43241b4ag BCM43241B4
+					strcpy(&dst[i+1], nv_name[5]);
 						break;
-//		case BCM43341_CHIP_ID: //bcm43341b0ag BCM43341B0 
-//					strcpy(&dst[i+1], nv_name[0]);
-//						break;
-//		case BCM4324_CHIP_ID: //bcm43241b4ag BCM43241B4
-//					strcpy(&dst[i+1], nv_name[0]);
-//						break;
 		case BCM4335_CHIP_ID: //bcm4339a0ag BCM4335A0
 					strcpy(&dst[i+1], nv_name[3]);
 						break;
