diff -Naur RetroArch/input/input_joypad.c RetroArch.patch/input/input_joypad.c
--- RetroArch/input/input_joypad.c	2015-01-11 13:54:38.050151366 +0100
+++ RetroArch.patch/input/input_joypad.c	2015-01-11 13:57:50.586625615 +0100
@@ -127,9 +127,19 @@
  * Returns: true (1) if key was pressed, otherwise
  * false (0).
  **/
+static bool hack_l3, hack_r3;
 bool input_joypad_pressed(const rarch_joypad_driver_t *drv,
       unsigned port, const struct retro_keybind *binds, unsigned key)
 {
+   if (key == RETRO_DEVICE_ID_JOYPAD_L3)
+      hack_l3 = input_joypad_is_pressed(drv, port, binds, key);
+
+   if (key == RETRO_DEVICE_ID_JOYPAD_R3)
+      hack_r3 = input_joypad_is_pressed(drv, port, binds, key);
+
+   if (key == RARCH_MENU_TOGGLE && hack_r3 && hack_l3)
+      return true;
+
    if (!drv || !binds[key].valid || 
          !input_joypad_is_pressed(drv, port, binds, key))
       return false;
