diff -Naur retroarch.git/menu/menu_displaylist.c retroarch.patch/menu/menu_displaylist.c
--- retroarch.git/menu/menu_displaylist.c	2015-05-13 02:23:34.961153683 +0200
+++ retroarch.patch/menu/menu_displaylist.c	2015-05-13 02:27:48.983130948 +0200
@@ -487,7 +487,7 @@
 
    menu_list_clear(list);
 
-   menu_entries_push_horizontal_menu_list_cores(list, info, settings->core_assets_directory, true);
+   menu_entries_push_horizontal_menu_list_cores(list, info, settings->menu_content_directory, true);
 
    menu_list_populate_generic(list, path, label, type, true);
 
