diff -Naur retroarch/menu/drivers_display/lakka.c retroarch.patch/menu/drivers_display/lakka.c
--- retroarch/menu/drivers_display/lakka.c	2015-03-07 19:57:54.520973734 +0100
+++ retroarch.patch/menu/drivers_display/lakka.c	2015-03-07 20:00:25.672479321 +0100
@@ -591,7 +591,7 @@
    active_item = (menu_item_t*)
       &active_category->items[active_category->active_item];
 
-   update_tweens(0.002);
+   update_tweens(driver.menu->dt / IDEAL_DT);
 
 #if defined(HAVE_FBO) && defined(LAKKA_EFFECTS)
    glBindFramebuffer(GL_FRAMEBUFFER, lakka->fbo);
diff -Naur retroarch/menu/drivers_display/lakka.h retroarch.patch/menu/drivers_display/lakka.h
--- retroarch/menu/drivers_display/lakka.h	2015-03-07 19:57:54.520973734 +0100
+++ retroarch.patch/menu/drivers_display/lakka.h	2015-03-07 19:59:39.859416298 +0100
@@ -25,7 +25,7 @@
 #define THEME "monochrome" // flatui or monochrome themes are available
 
 #ifndef LAKKA_DELAY
-#define LAKKA_DELAY 0.02
+#define LAKKA_DELAY 10
 #endif
 
 enum
diff -Naur retroarch/menu/menu.c retroarch.patch/menu/menu.c
--- retroarch/menu/menu.c	2015-03-07 19:57:54.492973086 +0100
+++ retroarch.patch/menu/menu.c	2015-03-07 20:08:08.395211872 +0100
@@ -20,6 +20,7 @@
 #include "../dynamic.h"
 #include "../frontend/frontend.h"
 #include "../../retroarch.h"
+#include "../../performance.h"
 #include <file/file_path.h>
 
 /**
@@ -436,6 +437,14 @@
    if (!driver.menu)
       return -1;
 
+   driver.menu->cur_time = rarch_get_time_usec();
+   driver.menu->dt = driver.menu->cur_time - driver.menu->old_time;
+   if (driver.menu->dt >= IDEAL_DT * 4)
+      driver.menu->dt = IDEAL_DT * 4;
+   if (driver.menu->dt <= IDEAL_DT / 4)
+      driver.menu->dt = IDEAL_DT / 4;
+   driver.menu->old_time = driver.menu->cur_time;
+
    driver.retro_ctx.poll_cb();
 
    if (input & input_repeat)
@@ -465,7 +474,7 @@
 
    driver.menu->mouse.enable = g_settings.menu.mouse_enable;
 
-   driver.menu->delay_count++;
+   driver.menu->delay_count += driver.menu->dt / IDEAL_DT;
 
    if (driver.block_input)
       trigger_input = 0;
diff -Naur retroarch/menu/menu_driver.h retroarch.patch/menu/menu_driver.h
--- retroarch/menu/menu_driver.h	2015-03-07 19:57:54.508973457 +0100
+++ retroarch.patch/menu/menu_driver.h	2015-03-07 20:06:44.549267343 +0100
@@ -23,6 +23,7 @@
 #include <retro_miscellaneous.h>
 #include "menu_list.h"
 #include "../settings_list.h"
+#include "../../libretro.h"
 
 #ifdef __cplusplus
 extern "C" {
@@ -70,9 +71,14 @@
 {
    void *userdata;
 
+   /* Delta timing */
+   float dt;
+   retro_time_t cur_time;
+   retro_time_t old_time;
+
    /* Used for key repeat */
-   unsigned delay_timer;
-   unsigned delay_count;
+   float delay_timer;
+   float delay_count;
 
    unsigned width;
    unsigned height;
diff -Naur retroarch/menu/menu.h retroarch.patch/menu/menu.h
--- retroarch/menu/menu.h	2015-03-07 19:57:54.508973457 +0100
+++ retroarch.patch/menu/menu.h	2015-03-07 20:04:54.726718994 +0100
@@ -46,6 +46,9 @@
 #define MENU_SETTINGS_CORE_OPTION_NONE  0xffff
 #define MENU_SETTINGS_CORE_OPTION_START 0x10000
 
+#ifndef IDEAL_DT
+#define IDEAL_DT (1.0 / 60.0 * 1000000.0)
+#endif
 
 #define MENU_KEYBOARD_BIND_TIMEOUT_SECONDS 5
 
