diff -Naur retroarch.git/command_event.c retroarch.patch/command_event.c
--- retroarch.git/command_event.c	2015-06-03 20:20:04.674299084 +0200
+++ retroarch.patch/command_event.c	2015-06-03 20:24:40.336796170 +0200
@@ -1333,6 +1333,11 @@
       case EVENT_CMD_QUIT_RETROARCH:
          rarch_main_set_state(RARCH_ACTION_STATE_FORCE_QUIT);
          break;
+      case EVENT_CMD_SHUTDOWN:
+         rarch_main_msg_queue_push("Shutting down...", 1, 180, true);
+         rarch_main_set_state(RARCH_ACTION_STATE_FORCE_QUIT);
+         system("shutdown -P now");
+         break;
       case EVENT_CMD_RESUME:
          rarch_main_set_state(RARCH_ACTION_STATE_MENU_RUNNING_FINISHED);
          break;
diff -Naur retroarch.git/command_event.h retroarch.patch/command_event.h
--- retroarch.git/command_event.h	2015-05-20 08:40:06.845167627 +0200
+++ retroarch.patch/command_event.h	2015-06-03 20:25:18.425693972 +0200
@@ -111,6 +111,8 @@
    EVENT_CMD_RESTART_RETROARCH,
    /* Force-quit RetroArch. */
    EVENT_CMD_QUIT_RETROARCH,
+   /* Shutdown the OS. */
+   EVENT_CMD_SHUTDOWN,
    /* Resume RetroArch when in menu. */
    EVENT_CMD_RESUME,
    /* Toggles pause. */
diff -Naur retroarch.git/menu/menu_setting.c retroarch.patch/menu/menu_setting.c
--- retroarch.git/menu/menu_setting.c	2015-06-03 20:20:04.686299366 +0200
+++ retroarch.patch/menu/menu_setting.c	2015-06-03 20:27:02.344140541 +0200
@@ -3583,12 +3583,19 @@
    /* Apple rejects iOS apps that lets you forcibly quit an application. */
    CONFIG_ACTION(
          "quit_retroarch",
-         "Quit RetroArch",
+         "Restart RetroArch",
          group_info.name,
          subgroup_info.name);
    settings_list_current_add_cmd(list, list_info, EVENT_CMD_QUIT_RETROARCH);
 #endif
 
+   CONFIG_ACTION(
+         "shutdown",
+         "Shutdown",
+         group_info.name,
+         subgroup_info.name);
+   settings_list_current_add_cmd(list, list_info, EVENT_CMD_SHUTDOWN);
+
    END_SUB_GROUP(list, list_info);
    END_GROUP(list, list_info);
 
