diff -Naur retroarch.git/libretro-common/file/file_path.c retroarch.patch/libretro-common/file/file_path.c
--- retroarch.git/libretro-common/file/file_path.c	2015-03-12 22:19:55.856710512 +0100
+++ retroarch.patch/libretro-common/file/file_path.c	2015-03-12 22:20:51.161993115 +0100
@@ -169,7 +169,7 @@
    if (strcmp(file_ext,"7z") == 0)
       return true;
 #endif
-#ifdef HAVE_ZLIB
+#if 0
    if (strcmp(file_ext,"zip") == 0)
       return true;
 #endif
