diff -Naur retroarch.git/gfx/drivers_context/drm_egl_ctx.c retroarch.patch/gfx/drivers_context/drm_egl_ctx.c
--- retroarch.git/gfx/drivers_context/drm_egl_ctx.c	2015-11-27 17:52:55.202785319 +0100
+++ retroarch.patch/gfx/drivers_context/drm_egl_ctx.c	2015-11-27 17:59:22.080211897 +0100
@@ -285,7 +285,10 @@
    drm_free();
 
    if (g_drm_fd >= 0)
+   {
+      drmDropMaster(drm->g_drm_fd);
       retro_fclose(drm->g_drm);
+   }
 
    g_gbm_surface      = NULL;
    g_gbm_dev          = NULL;
@@ -368,6 +371,8 @@
    drm->g_fb_width  = g_drm_connector->modes[0].hdisplay;
    drm->g_fb_height = g_drm_connector->modes[0].vdisplay;
 
+   drmSetMaster(drm->g_drm_fd);
+
    g_gbm_dev        = gbm_create_device(fd);
 
    if (!g_gbm_dev)
