diff --git a/arch/arm/mach-meson8/pm.c b/arch/arm/mach-meson8/pm.c
index 3f4d04b..883421f 100755
--- a/arch/arm/mach-meson8/pm.c
+++ b/arch/arm/mach-meson8/pm.c
@@ -346,11 +346,13 @@ static void m6ref_set_vccx2(int power_on)
     if(power_on == OFF) {
         printk("m6ref_set_vccx2: OFF");
         CLEAR_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<15);
-        SET_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
+        //SET_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
+        CLEAR_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
     } else {
         printk("m6ref_set_vccx2: ON");
         CLEAR_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<15);
-        CLEAR_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
+        //CLEAR_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
+        SET_AOBUS_REG_MASK(AO_GPIO_O_EN_N, 1<<31);
     }
     return;
 }
