diff -Naur retroarch.git/gfx/drivers/gl.c retroarch.patch/gfx/drivers/gl.c
--- retroarch.git/gfx/drivers/gl.c	2015-03-25 18:23:48.226216528 +0100
+++ retroarch.patch/gfx/drivers/gl.c	2015-03-25 23:20:21.455106006 +0100
@@ -1867,7 +1867,7 @@
    unsigned gles_major = 0, gles_minor = 0;
 
    /* There are both APPLE and EXT variants. */
-   if (gl_query_extension(gl, "BGRA8888"))
+   if (0)//(gl_query_extension(gl, "BGRA8888"))
       RARCH_LOG("[GL]: BGRA8888 extension found for GLES.\n");
    else
    {
