diff --git a/driver.c b/driver.c
index 5d4eeaf..e8a2c14 100644
--- a/driver.c
+++ b/driver.c
@@ -339,6 +339,7 @@ bool driver_update_system_av_info(const struct retro_system_av_info *info)
 void init_drivers(int flags)
 {
    driver_t *driver = driver_get_ptr();
+   global_t *global = global_get_ptr();
 
    if (flags & DRIVER_VIDEO)
       driver->video_data_own = false;
@@ -391,7 +392,7 @@ void init_drivers(int flags)
    if (flags & DRIVER_MENU)
    {
       init_menu();
-      menu_driver_context_reset();
+      menu_update_libretro_info(&global->menu.info);
    }
 #endif
 
diff --git a/menu/menu.c b/menu/menu.c
index 6804600..222356f 100644
--- a/menu/menu.c
+++ b/menu/menu.c
@@ -71,7 +71,7 @@ static void draw_frame(void)
  *
  * Update menu state which depends on config.
  **/
-static void menu_update_libretro_info(struct retro_system_info *info)
+void menu_update_libretro_info(struct retro_system_info *info)
 {
 #ifndef HAVE_DYNAMIC
    retro_get_system_info(info);
@@ -167,8 +167,6 @@ bool menu_load_content(void)
       return false;
    }
 
-   menu_update_libretro_info(&global->menu.info);
-
    menu_shader_manager_init(driver->menu);
 
    rarch_main_command(RARCH_CMD_HISTORY_INIT);
diff --git a/menu/menu.h b/menu/menu.h
index 9f87cd6..a457cd0 100644
--- a/menu/menu.h
+++ b/menu/menu.h
@@ -199,6 +199,8 @@ void menu_apply_deferred_settings(void);
 
 bool menu_display_update_pending(void);
 
+void menu_update_libretro_info(struct retro_system_info *info);
+
 #ifdef __cplusplus
 }
 #endif
