diff -Naur RetroArch/frontend/menu/menu_common.c RetroArch.patch/frontend/menu/menu_common.c
--- RetroArch/frontend/menu/menu_common.c	2014-03-27 17:00:11.577434728 +0100
+++ RetroArch.patch/frontend/menu/menu_common.c	2014-03-27 17:58:56.498440172 +0100
@@ -545,7 +545,7 @@
 #endif
    }
 
-   input_state |= input_key_pressed_func(RARCH_MENU_TOGGLE) ? (1ULL << RARCH_MENU_TOGGLE) : 0;
+   input_state |= input_key_pressed_func(RARCH_MENU_TOGGLE) || (input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_L3) && input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_R3)) ? (1ULL << RARCH_MENU_TOGGLE) : 0;
 
    input_pop_analog_dpad((struct retro_keybind*)binds[0]);
    for (i = 0; i < MAX_PLAYERS; i++)
diff -Naur RetroArch/retroarch.c RetroArch.patch/retroarch.c
--- RetroArch/retroarch.c	2014-03-27 17:00:11.653436474 +0100
+++ RetroArch.patch/retroarch.c	2014-03-27 17:57:12.232040870 +0100
@@ -3097,7 +3097,7 @@
    static bool old_rmenu_toggle = true;
 
    // Always go into menu if dummy core is loaded.
-   bool rmenu_toggle = input_key_pressed_func(RARCH_MENU_TOGGLE) || (g_extern.libretro_dummy && !old_rmenu_toggle);
+   bool rmenu_toggle = input_key_pressed_func(RARCH_MENU_TOGGLE) || (input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_L3) && input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_R3)) || (g_extern.libretro_dummy && !old_rmenu_toggle);
    if (rmenu_toggle && !old_rmenu_toggle)
    {
       g_extern.lifecycle_state |= (1ULL << MODE_MENU_PREINIT);
