diff -Naur RetroArch/frontend/menu/menu_input_line_cb.c RetroArch.patch/frontend/menu/menu_input_line_cb.c
--- RetroArch/frontend/menu/menu_input_line_cb.c	2014-08-26 09:15:55.919769986 +0200
+++ RetroArch.patch/frontend/menu/menu_input_line_cb.c	2014-08-26 09:44:47.900901874 +0200
@@ -325,6 +325,8 @@
       }
    }
    input_state |= input_key_pressed_func(RARCH_MENU_TOGGLE)
+      || (input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_L3)
+      && input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_R3))
       ? (1ULL << RARCH_MENU_TOGGLE) : 0;
 
    input_pop_analog_dpad((struct retro_keybind*)binds[0]);
diff -Naur RetroArch/retroarch.c RetroArch.patch/retroarch.c
--- RetroArch/retroarch.c	2014-08-26 09:15:55.995771764 +0200
+++ RetroArch.patch/retroarch.c	2014-08-26 09:45:59.970585927 +0200
@@ -3123,6 +3123,8 @@
 {
    static bool old_rmenu_toggle = true;
    bool rmenu_toggle            = input_key_pressed_func(RARCH_MENU_TOGGLE)
+      || (input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_L3)
+      && input_key_pressed_func(RETRO_DEVICE_ID_JOYPAD_R3))
       || (g_extern.libretro_dummy && !old_rmenu_toggle);
 
    /* Always go into menu if dummy core is loaded. */
