diff -Nur a/Makefile b/Makefile
--- a/Makefile	2016-03-18 12:09:19.000000000 +0100
+++ b/Makefile	2016-05-11 17:48:47.086770207 +0200
@@ -39,7 +39,7 @@
 CONFIG_GSPI_HCI = n
 ########################## Features ###########################
 CONFIG_MP_INCLUDED = y
-CONFIG_POWER_SAVING = y
+CONFIG_POWER_SAVING = n
 CONFIG_USB_AUTOSUSPEND = n
 CONFIG_HW_PWRP_DETECTION = n
 CONFIG_WIFI_TEST = n
@@ -57,7 +57,7 @@
 CONFIG_SIGNAL_SCALE_MAPPING = n
 CONFIG_80211W = n
 CONFIG_REDUCE_TX_CPU_LOADING = n
-CONFIG_BR_EXT = y
+CONFIG_BR_EXT = n
 CONFIG_ANTENNA_DIVERSITY = n
 CONFIG_TDLS = n
 CONFIG_WIFI_MONITOR = n
@@ -70,9 +70,9 @@
 CONFIG_PNO_SET_DEBUG = n
 CONFIG_AP_WOWLAN = n
 ######### Notify SDIO Host Keep Power During Syspend ##########
-CONFIG_RTW_SDIO_PM_KEEP_POWER = y
+CONFIG_RTW_SDIO_PM_KEEP_POWER = n
 ###################### Platform Related #######################
-CONFIG_PLATFORM_I386_PC = y
+CONFIG_PLATFORM_I386_PC = n
 CONFIG_PLATFORM_ANDROID_X86 = n
 CONFIG_PLATFORM_ARM_NV_TK1 = n
 CONFIG_PLATFORM_ANDROID_INTEL_X86 = n
@@ -108,8 +108,7 @@
 CONFIG_PLATFORM_ARM_SUNxI = n
 CONFIG_PLATFORM_ARM_SUN6I = n
 CONFIG_PLATFORM_ARM_SUN7I = n
-CONFIG_PLATFORM_ARM_SUN8I_W3P1 = n
-CONFIG_PLATFORM_ARM_SUN8I_W5P1 = n
+CONFIG_PLATFORM_ARM_SUN8I = y
 CONFIG_PLATFORM_ACTIONS_ATM702X = n
 CONFIG_PLATFORM_ACTIONS_ATV5201 = n
 CONFIG_PLATFORM_ACTIONS_ATM705X = n
@@ -1368,13 +1367,11 @@
 KSRC := /home/android_sdk/Allwinner/a20/kitkat-a20_v4.4/lichee/linux-3.4
 endif
 
-ifeq ($(CONFIG_PLATFORM_ARM_SUN8I_W3P1), y)
+ifeq ($(CONFIG_PLATFORM_ARM_SUN8I), y)
 EXTRA_CFLAGS += -DCONFIG_LITTLE_ENDIAN
 EXTRA_CFLAGS += -DCONFIG_PLATFORM_ARM_SUN8I
-EXTRA_CFLAGS += -DCONFIG_PLATFORM_ARM_SUN8I_W3P1
 EXTRA_CFLAGS += -DCONFIG_TRAFFIC_PROTECT
 # default setting for Android 4.1, 4.2
-EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
 EXTRA_CFLAGS += -DCONFIG_P2P_IPS
 
diff -Nur a/platform/platform_ARM_SUNnI_sdio.c b/platform/platform_ARM_SUNnI_sdio.c
--- a/platform/platform_ARM_SUNnI_sdio.c	2016-03-18 12:09:19.000000000 +0100
+++ b/platform/platform_ARM_SUNnI_sdio.c	2016-05-11 17:46:48.340097578 +0200
@@ -41,12 +41,6 @@
 extern void sunxi_mci_rescan_card(unsigned id, unsigned insert);
 #endif
 
-#ifdef CONFIG_PLATFORM_ARM_SUN8I_W5P1
-extern int get_rf_mod_type(void);
-#else
-extern int wifi_pm_get_mod_type(void);
-#endif
-
 extern void wifi_pm_power(int on);
 #ifdef CONFIG_GPIO_WAKEUP
 extern unsigned int oob_irq;
@@ -67,11 +61,11 @@
 	script_item_u val;
 	script_item_value_type_e type;
 
-#ifdef CONFIG_PLATFORM_ARM_SUN8I_W5P1
-	unsigned int mod_sel = get_rf_mod_type();
-#else
-	unsigned int mod_sel = wifi_pm_get_mod_type();
-#endif
+	type = script_get_item("wifi_para", "wifi_used", &val);
+	if ((SCIRPT_ITEM_VALUE_TYPE_INT != type) || (0 == val.val)) {
+		DBG_871X("wifi is not used\n");
+		return -1;
+	}
 
 	type = script_get_item("wifi_para", "wifi_sdc_id", &val);
 	if (SCIRPT_ITEM_VALUE_TYPE_INT!=type) {
@@ -79,7 +73,7 @@
 		ret = -1;
 	} else {
 		sdc_id = val.val;
-		DBG_871X("----- %s sdc_id: %d, mod_sel: %d\n", __FUNCTION__, sdc_id, mod_sel);
+		DBG_871X("----- %s sdc_id: %d\n", __FUNCTION__, sdc_id);
 
 #if defined(CONFIG_PLATFORM_ARM_SUN6I) || defined(CONFIG_PLATFORM_ARM_SUN7I)
 		sw_mci_rescan_card(sdc_id, 1);
@@ -93,16 +87,7 @@
 	}
 
 #ifdef CONFIG_GPIO_WAKEUP
-#ifdef CONFIG_PLATFORM_ARM_SUN8I_W5P1
 	type = script_get_item("wifi_para", "wl_host_wake", &val);
-#else
-#ifdef CONFIG_RTL8723B
-	type = script_get_item("wifi_para", "rtl8723bs_wl_host_wake", &val);
-#endif
-#ifdef CONFIG_RTL8188E
-	type = script_get_item("wifi_para", "rtl8189es_host_wake", &val);
-#endif
-#endif /* CONFIG_PLATFORM_ARM_SUN8I_W5P1 */
 	if (SCIRPT_ITEM_VALUE_TYPE_PIO != type) {
 		DBG_871X("No definition of wake up host PIN\n");
 		ret = -1;
