diff -Nru a/update b/update
--- a/update	1970-01-01 03:00:00.000000000 +0300
+++ b/update	2014-12-29 09:19:53.768905850 +0300
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+IMAGE="/tmp/cache/update.zip"
+source=$1
+
+mkdir -p /tmp/cache
+mount -t ext4 /dev/cache /tmp/cache
+
+wget $source -O /tmp/cache/update.zip
+sync
+
+if [ ! -f "${IMAGE}" ]; then
+  echo "Error: ${IMAGE} not found"
+  exit 1
+fi
+
+echo -e "--update_package=/cache/update.zip\n--wipe_cache" > /tmp/cache/recovery/command || exit 1
+
+umount /tmp/cache
+#mount -o,ro remount /storage
+
+echo "Update firmware"
+reboot recovery
+
