diff -Naur linux-5df008c.orig/drivers/amlogic/ionvideo/ionvideo.c linux-5df008c/drivers/amlogic/ionvideo/ionvideo.c
--- linux-5df008c.orig/drivers/amlogic/ionvideo/ionvideo.c	2016-07-12 18:59:37.761568802 +0200
+++ linux-5df008c/drivers/amlogic/ionvideo/ionvideo.c	2016-07-12 19:01:16.000000000 +0200
@@ -941,7 +941,7 @@
     q->ops = &ionvideo_video_qops;
     q->mem_ops = &vb2_ion_memops;
     //q->timestamp_type = V4L2_BUF_FLAG_TIMESTAMP_MONOTONIC;
-	q->timestamp_flags = V4L2_BUF_FLAG_TIMESTAMP_MONOTONIC;
+    q->timestamp_flags = V4L2_BUF_FLAG_TIMESTAMP_MONOTONIC;
 
     ret = vb2_queue_init(q);
     if (ret)
@@ -989,7 +989,7 @@
 
 static u32 buf_num_queue(struct vb2_queue *q)
 {
-    return atomic_read(&q->queued_count);
+    return atomic_read(&q->owned_by_drv_count);
 }
 
 static ssize_t vframe_states_show(struct class *class, struct class_attribute* attr, char* buf)
