diff -Naur systemd-218/src/journal/journal-file.c systemd-218.patch/src/journal/journal-file.c
--- systemd-218/src/journal/journal-file.c	2014-12-08 19:38:55.767371110 +0100
+++ systemd-218.patch/src/journal/journal-file.c	2015-01-18 01:23:21.391634433 +0100
@@ -26,6 +26,9 @@
 #include <sys/statvfs.h>
 #include <fcntl.h>
 #include <stddef.h>
+#define __USE_KERNEL_XATTR_DEFS
+#define XATTR_CREATE 1
+#define XATTR_REPLACE 2
 #include <sys/xattr.h>
 
 #include "journal-def.h"
