From 93b2d3e1001fb4986f820ec6c50575a6a202ad41 Mon Sep 17 00:00:00 2001
From: demetris <ierokipides.dem@gmail.com>
Date: Thu, 24 Nov 2016 01:23:07 +0200
Subject: [PATCH] Fix for corrupted image at kernel boot.

---
 drivers/video/sunxi_display.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/video/sunxi_display.c b/drivers/video/sunxi_display.c
index 50b16a9..ff98c95 100644
--- a/drivers/video/sunxi_display.c
+++ b/drivers/video/sunxi_display.c
@@ -18,6 +18,7 @@
 #include <asm/io.h>
 #include <axp_pmic.h>
 #include <errno.h>
+#include <malloc.h>
 #include <fdtdec.h>
 #include <fdt_support.h>
 #include <i2c.h>
@@ -1488,8 +1489,7 @@ void *video_hw_init(void)
 	       sunxi_get_mon_desc(sunxi_display.monitor),
 	       overscan_x, overscan_y);
 
-	gd->fb_base = gd->bd->bi_dram[0].start +
-		      gd->bd->bi_dram[0].size - sunxi_display.fb_size;
+	gd->fb_base = (unsigned long)memalign(0x1000, sunxi_display.fb_size);
 	sunxi_engines_init();
 
 	fb_dma_addr = gd->fb_base - CONFIG_SYS_SDRAM_BASE;
-- 
2.7.4

