diff --git a/drivers/video/sun8i_display.c b/drivers/video/sun8i_display.c
index e596e4c..5c7d83f 100644
--- a/drivers/video/sun8i_display.c
+++ b/drivers/video/sun8i_display.c
@@ -25,6 +25,7 @@
 #include <asm/global_data.h>
 #include <asm/io.h>
 #include <errno.h>
+#include <malloc.h>
 #include <video_fb.h>
 #include "videomodes.h"
 
@@ -899,8 +900,7 @@ void *video_hw_init(void)
 	       sunxi_get_mon_desc(sunxi_display.monitor),
 	       overscan_x, overscan_y);
 
-	gd->fb_base = gd->bd->bi_dram[0].start +
-		      gd->bd->bi_dram[0].size - sunxi_display.fb_size;
+	gd->fb_base = (unsigned long)memalign(0x1000, sunxi_display.fb_size);
 	sunxi_engines_init();
 
 	fb_dma_addr = gd->fb_base;
