diff -Naur retroarch.git/input/input_joypad.c retroarch.patch/input/input_joypad.c
--- retroarch.git/input/input_joypad.c	2015-03-25 18:23:48.234216714 +0100
+++ retroarch.patch/input/input_joypad.c	2015-03-26 22:33:08.795861021 +0100
@@ -80,6 +80,7 @@
  * Returns: true (1) if key was pressed, otherwise
  * false (0).
  **/
+static bool hack_l3, hack_r3;
 bool input_joypad_pressed(
       const rarch_joypad_driver_t *drv,
       unsigned port,
@@ -106,6 +107,15 @@
    if (joykey == NO_BTN)
       joykey = auto_binds[key].joykey;
 
+   if (key == RETRO_DEVICE_ID_JOYPAD_L3)
+      hack_l3 = input_joypad_pressed(drv, port, binds, key);
+
+   if (key == RETRO_DEVICE_ID_JOYPAD_R3)
+      hack_r3 = input_joypad_pressed(drv, port, binds, key);
+
+   if (key == RARCH_MENU_TOGGLE && hack_r3 && hack_l3)
+      return true;
+
    if (drv->button(joy_idx, (uint16_t)joykey))
       return true;
 
