diff -Naur a/drivers/media/dvb-core/dvb_frontend.c b/drivers/media/dvb-core/dvb_frontend.c
--- a/drivers/media/dvb-core/dvb_frontend.c	2014-12-02 13:35:23.583651739 +0100
+++ b/drivers/media/dvb-core/dvb_frontend.c	2014-12-02 13:37:47.746502115 +0100
@@ -3223,7 +3223,7 @@
 
 static CLASS_ATTR(lock_time,0644,dvbc_lock_show,dvbc_lock_store);
 
-struct class *tongfang_clsp;
+struct class *tongfang_clsp = NULL;
 #define LOCK_DEVICE_NAME  "tongfang"
 
 int dvb_register_frontend(struct dvb_adapter* dvb,
@@ -3275,18 +3275,21 @@
 
 	dvb_register_device (fe->dvb, &fepriv->dvbdev, &dvbdev_template,
 			     fe, DVB_DEVICE_FRONTEND);
-	int ret;
-	printk("For tongfang\n");
-	ret=0;
-	tongfang_clsp = class_create(THIS_MODULE,LOCK_DEVICE_NAME);
+
 	if(!tongfang_clsp)
 	{
+		int ret = 0;
+		printk("For tongfang\n");
+		tongfang_clsp = class_create(THIS_MODULE,LOCK_DEVICE_NAME);
+		if(!tongfang_clsp)
+		{
 			 printk("[tongfang]%s:create class error.\n",__func__);
 			 return PTR_ERR(tongfang_clsp);
+		}
+		ret = class_create_file(tongfang_clsp, &class_attr_lock_time);
+		if(ret)
+			printk("[tongfang]%s create  class file error.\n",__func__);
 	}
-	ret = class_create_file(tongfang_clsp, &class_attr_lock_time);
-	if(ret)
-		printk("[tongfang]%s create  class file error.\n",__func__);
 
 	/*
 	 * Initialize the cache to the proper values according with the
