--- a/common.h.orig	2016-02-07 14:33:17.519832421 +0100
+++ b/common.h	2016-02-07 14:33:20.704863567 +0100
@@ -32,18 +32,17 @@
 
 #define SECTION_FILTER_TABLE_SIZE       5
 
-#define ERROR_IF_FUNC(exp, errstr, func, ret)                       \
-  do {                                                              \
-     if (exp) {                                                     \
-        char tmp[64];                                               \
-        esyslog("[%s,%d]: "errstr": %s", __FILE__, __LINE__,        \
-                strerror_r(errno, tmp, sizeof(tmp)));               \
-        func;                                                       \
-        ret;                                                        \
-        }                                                           \
+#define ERROR_IF_FUNC(exp, errstr, func, ret)                  \
+  do {                                                         \
+     if (exp) {                                                \
+        char tmp[64];                                          \
+        esyslog("[%s,%d]: " errstr ": %s", __FILE__, __LINE__, \
+                strerror_r(errno, tmp, sizeof(tmp)));          \
+        func;                                                  \
+        ret;                                                   \
+        }                                                      \
   } while (0)
 
-
 #define ERROR_IF_RET(exp, errstr, ret) ERROR_IF_FUNC(exp, errstr, ,ret);
 
 #define ERROR_IF(exp, errstr) ERROR_IF_FUNC(exp, errstr, , );
