diff -Naur lima-drm/Makefile.inc lima-drm.patch/Makefile.inc
--- lima-drm/Makefile.inc	2014-06-19 17:10:48.586008086 +0200
+++ lima-drm.patch/Makefile.inc	2014-06-19 18:50:23.909233996 +0200
@@ -5,10 +5,10 @@
 
 CFLAGS ?= -O0 -g -Wall
 
-CROSS_COMPILE ?=
-CC = $(CROSS_COMPILE)gcc
-LD = $(CROSS_COMPILE)ld
-AS = $(CROSS_COMPILE)as
+#CROSS_COMPILE ?=
+#CC = $(CROSS_COMPILE)gcc
+#LD = $(CROSS_COMPILE)ld
+#AS = $(CROSS_COMPILE)as
 
 triplet = $(shell $(CC) -dumpmachine)
 ifeq ($(triplet), arm-linux-androideabi)
