diff -Naur lima-drm/limare/lib/Makefile lima-drm.patch/limare/lib/Makefile
--- lima-drm/limare/lib/Makefile	2014-06-19 17:10:48.586008086 +0200
+++ lima-drm.patch/limare/lib/Makefile	2014-06-19 18:58:37.920670300 +0200
@@ -2,7 +2,7 @@
 
 include $(TOP)Makefile.inc
 
-CFLAGS += -I$(TOP)include -I/usr/include/libdrm -I/usr/include/exynos -fPIC
+CFLAGS += -I$(TOP)include #-I/usr/include/libdrm -I/usr/include/exynos -fPIC
 ASFLAGS += -mfpu=neon -mfloat-abi=hard -fPIC
 
 all: liblimare.so
