diff --git a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
index 771364b..a6c4076 100644
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
+++ b/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
@@ -38,6 +38,7 @@
 #include "settings/AdvancedSettings.h"
 #if defined(HAS_LIBAMCODEC)
 #include "utils/AMLUtils.h"
+#include "utils/SysfsUtils.h"
 #endif
 
 
@@ -539,6 +540,25 @@ bool CAESinkALSA::Initialize(AEAudioFormat &format, std::string &device)
   {
     aml_set_audio_passthrough(m_passthrough);
     device = "default";
+
+    int aml_digital_codec;
+    switch(format.m_dataFormat)
+    {
+      case AE_FMT_DTSHD:
+        aml_digital_codec = 5;
+        break;
+
+      case AE_FMT_TRUEHD:
+        aml_digital_codec = 7;
+        break;
+
+      default:
+        if (inconfig.channels > 2)
+          aml_digital_codec = 6;
+        else
+          aml_digital_codec = 0;
+    }
+    SysfsUtils::SetInt("/sys/class/audiodsp/digital_codec", aml_digital_codec);
   }
 #endif
 
