diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 42d9832..c9b7ac6 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -263,6 +263,11 @@ KeyMap keyMap[] = {
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
+  { KEY_SELECT        , XBMCK_RETURN      },
+  { KEY_RED           , XBMCK_TAB         },
+  { KEY_GREEN         , XBMCK_z           },
+  { KEY_YELLOW        , XBMCK_i           },
+  { KEY_BLUE          , XBMCK_c           },
   { KEY_MAIL          , XBMCK_LAUNCH_MAIL },
   { KEY_SEARCH        , XBMCK_BROWSER_SEARCH},
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
