diff --git a/xbmc/input/linux/LinuxInputDevices.cpp b/xbmc/input/linux/LinuxInputDevices.cpp
index 0b0f7ae..c84930f 100644
--- a/xbmc/input/linux/LinuxInputDevices.cpp
+++ b/xbmc/input/linux/LinuxInputDevices.cpp
@@ -265,6 +265,11 @@ KeyMap keyMap[] = {
   { KEY_YELLOW        , XBMCK_YELLOW      },
   { KEY_BLUE          , XBMCK_BLUE        },
   { KEY_HOMEPAGE      , XBMCK_HOMEPAGE    },
+  { KEY_SELECT        , XBMCK_RETURN      },
+  { KEY_RED           , XBMCK_TAB         },
+  { KEY_GREEN         , XBMCK_z           },
+  { KEY_YELLOW        , XBMCK_i           },
+  { KEY_BLUE          , XBMCK_c           },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
