diff -Nru a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp
--- a/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp	2014-12-09 22:29:46.000000000 +0300
+++ b/xbmc/cores/dvdplayer/DVDCodecs/Video/AMLCodec.cpp	2014-12-22 16:51:53.067246389 +0300
@@ -1581,14 +1581,14 @@
         am_private->gcodec.param = (void*)(EXTERNAL_PTS | SYNC_OUTSIDE);
       break;
     case VFORMAT_H264_4K2K:
-      if (aml_get_device_type() == AML_DEVICE_TYPE_M8) {
+      if ((aml_get_device_type() == AML_DEVICE_TYPE_M8) || (aml_get_device_type() == AML_DEVICE_TYPE_M8M2)) {
         am_private->gcodec.format = VIDEO_DEC_FORMAT_H264_4K2K;
         am_private->gcodec.param  = (void*)EXTERNAL_PTS;
         // h264 in an avi file
         if (m_hints.ptsinvalid)
           am_private->gcodec.param = (void*)(EXTERNAL_PTS | SYNC_OUTSIDE);
       } else {
-        CLog::Log(LOGDEBUG, "CAMLCodec::OpenDecoder codec init failed, 4K supported only on Meson8.");
+        CLog::Log(LOGDEBUG, "CAMLCodec::OpenDecoder codec init failed, 4K supported only on Meson8 and Meson8m2.");
         return false;
       }
       break; 
