diff -Naur wpa_supplicant-2.6/src/crypto/crypto_openssl.c wpa_supplicant-2.6.patch/src/crypto/crypto_openssl.c
--- wpa_supplicant-2.6/src/crypto/crypto_openssl.c	2016-10-02 20:51:11.000000000 +0200
+++ wpa_supplicant-2.6.patch/src/crypto/crypto_openssl.c	2016-10-13 09:29:24.874443700 +0200
@@ -611,7 +611,7 @@
 
 void * dh5_init(struct wpabuf **priv, struct wpabuf **publ)
 {
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 	DH *dh;
 	struct wpabuf *pubkey = NULL, *privkey = NULL;
 	size_t publen, privlen;
