--- /drivers/net/ethernet/sunxi/eth/Kconfig	2015-01-27 04:29:28.000000000 +0200
+++ /drivers/net/ethernet/sunxi/eth/Kconfig	2015-05-27 14:55:13.404754155 +0300
@@ -35,5 +35,13 @@
 	 If external PHY power is exist, and it want to be controled,
 	 select it.  If not, it mean the power of PHY already on.
 
+config GMAC_PHY_POWER
+        bool "External PHY power control"
+	depends on SUNXI_GETH
+	default y
+	---help---
+	 If external PHY power is exist, and it want to be controled,
+	 select it.  If not, it mean the power of PHY already on.
+	 
 endif
 
--- /drivers/net/ethernet/sunxi/eth/sunxi_geth.c	2015-01-27 04:29:28.000000000 +0200
+++ /drivers/net/ethernet/sunxi/eth/sunxi_geth.c	2015-05-27 15:14:49.264145949 +0300
@@ -154,6 +154,9 @@
 
 	spinlock_t lock;
 	spinlock_t tx_lock;
+#ifdef CONFIG_GMAC_PHY_POWER
+	u32 gpio_power_hd;
+#endif
 };
 
 #ifdef CONFIG_GETH_PHY_POWER
@@ -195,6 +198,11 @@
 static int geth_power_on(struct geth_priv *priv)
 {
 	int value;
+        
+#ifdef CONFIG_GMAC_PHY_POWER
+	gpio_set_value(priv->gpio_power_hd, 1);
+#endif
+        
 #ifdef CONFIG_GETH_PHY_POWER
 	struct regulator **regu;
 	int ret = 0, i = 0;
@@ -258,6 +266,11 @@
 static void geth_power_off(struct geth_priv *priv)
 {
 	int value;
+        
+#ifdef CONFIG_GMAC_PHY_POWER
+	gpio_set_value(priv->gpio_power_hd, 0);
+#endif
+        
 #ifdef CONFIG_GETH_PHY_POWER
 	struct regulator **regu = priv->power;
 	int i = 0;
@@ -1607,6 +1620,12 @@
 	struct geth_priv *priv = netdev_priv(ndev);
 	int ret = 0;
 	struct resource *res;
+        
+#ifdef CONFIG_GMAC_PHY_POWER
+	script_item_value_type_e  type;
+	script_item_u item;
+	int req_status;
+#endif
 
 	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "geth_extclk");
 	if (unlikely(!res)){
@@ -1681,6 +1700,22 @@
 		}
 	}
 #endif
+        
+#ifdef CONFIG_GMAC_PHY_POWER && CONFIG_GETH_SCRIPT_SYS
+	type = script_get_item("gmac_phy_power", "gmac_phy_power_en", &item);
+	if (SCIRPT_ITEM_VALUE_TYPE_PIO != type) {
+		pr_err("script_get_item return type err\n");
+		return -EFAULT;
+	}
+	/*request gpio*/
+	req_status = gpio_request(item.gpio.gpio, NULL);
+	if (0 != req_status) {
+		pr_err("request gpio failed!\n");
+	}
+	gpio_direction_output(item.gpio.gpio, 1);
+	priv->gpio_power_hd = item.gpio.gpio;        
+ #endif
+        
 	return 0;
 
 pin_err:
