diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
old mode 100755
new mode 100644
index 7e7df57..2df9ce1
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdhc.dtd
@@ -142,11 +142,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -157,7 +157,7 @@ void root_func(){
 		compatible = "amlogic,deinterlace";
 		dev_name = "deinterlace";
 		status = "ok";
-		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
+		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
 /// ***************************************************************************************
@@ -171,7 +171,7 @@ void root_func(){
 		dev_name = "mesonstream.0";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -188,6 +188,19 @@ void root_func(){
 //		reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+///	-	picture decoder
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";								
+		status = "okay";
+        share-memory-name = "vdec0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x04000000>; //64M
+	}; 
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
@@ -236,16 +249,16 @@ void root_func(){
 //$$ L2 PROP_U32 2 = "reg"
 //$$ L2 PROP_U32 = "irq"
 //$$ L2 PROP_U32 = "vdin_id"
-    vdin1{
-        compatible = "amlogic,vdin";
-        dev_name = "vdin1";
-        status = "ok";
-        reserve-memory = <0x01000000>;
-        miracast_size = <1920 1080>;//1920x1080x2x4=17M
-        reserve-iomap = "true";
-        irq = <117>;
-        vdin_id = <1>;
-    };
+//    vdin1{
+//        compatible = "amlogic,vdin";
+//        dev_name = "vdin1";
+//        status = "ok";
+//        reserve-memory = <0x01000000>;
+//	  miracast_size = <1920 1080>;//1920x1080x2x4=17M
+//	  reserve-iomap = "true";
+//        irq = <117>;
+//        vdin_id = <1>;
+//	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-AMLVIDEO2
@@ -253,13 +266,13 @@ void root_func(){
 //$$ DEVICE="amlvideo2"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2 ="reg"
-    amlvideo2{
-        compatible = "amlogic,amlvideo2";
-        dev_name = "amlvideo2.0";
-        status = "okay";
-        reserve-memory = <0x01800000>;
-        reserve-iomap = "true";
-    };
+//    amlvideo2{
+//		compatible = "amlogic,amlvideo2";
+//		dev_name = "amlvideo2.0";
+//		status = "okay";
+//		reserve-memory = <0x01800000>;
+//		reserve-iomap = "true";
+//	};
 
 /// ***************************************************************************************
 ///	-	ION
@@ -467,7 +480,7 @@ sdio{
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            // lhh gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -946,8 +959,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1441,13 +1454,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-			amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+		
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
@@ -1789,12 +1797,12 @@ sdio{
 //$$ L2 PROP_U32 8 = "amlogic,clrmask"
 //$$ L2 PROP_STR 12 = "amlogic,pins"
 		dvb_p_ts0_pins: dvb_p_ts0_pins {
-			amlogic,setmask = <3 0x37>;
+			amlogic,setmask = <3 0x3F>;
 			amlogic,clrmask = <1 0xCCE41
 								4 0x2000000
 								7 0x20000000
 								9 0x431F>;
-			amlogic,pins = "GPIOY_9","GPIOY_10","GPIOY_11","GPIOY_12","GPIOY_13","GPIOY_14","GPIOY_15","GPIOY_16","GPIOY_0","GPIOY_1","GPIOY_8";
+			amlogic,pins = "GPIOY_9","GPIOY_10","GPIOY_11","GPIOY_12","GPIOY_13","GPIOY_14","GPIOY_15","GPIOY_16","GPIOY_0","GPIOY_1","GPIOY_4","GPIOY_8";
 		};
 
 //$$ MATCH "dvb_s_ts0_pins_match" = "&dvb_s_ts0_pins"
@@ -1802,11 +1810,11 @@ sdio{
 //$$ L2 PROP_U32 6 = "amlogic,clrmask"
 //$$ L2 PROP_STR 5 = "amlogic,pins"
 		dvb_s_ts0_pins: dvb_s_ts0_pins {
-			amlogic,setmask = <3 0x17>;
+			amlogic,setmask = <3 0x1F>;
 			amlogic,clrmask = <1 0xCCE41
 								4 0x2000000
 								9 0x310>;
-			amlogic,pins = "GPIOY_9","GPIOY_0","GPIOY_1","GPIOY_8";
+			amlogic,pins = "GPIOY_9","GPIOY_0","GPIOY_1","GPIOY_4","GPIOY_8";
 		};
 
 //$$ MATCH "dvb_fe_pins_match" = "&dvb_fe_pins"
@@ -1820,14 +1828,6 @@ sdio{
 			amlogic,pins = "GPIOY_6";
 		};
 
-		smc_pins:smc_pins {
-		amlogic,setmask = <4 0xC0000>;
-		amlogic,clrmask = <4 0x3F0
-								3 0x1000>;
-		amlogic,pins= "GPIOX_18","GPIOX_19";
-		
-		};
-
 	};
 /// ***************************************************************************************
 ///	-	Ethernet
@@ -1890,36 +1890,4 @@ sdio{
 		pinctrl-0 = <&dvb_fe_pins>;
 	};
 
-	smartcard {
-		compatible = "amlogic,smartcard";
-		smc0_irq = <69>;
-		smc0_clock_source = <0>;
-		//smc0_enable_pin = "GPIOX_16";
-		//smc0_enable_level = <1>;
-		//smc0_5v3v_pin = "GPIOY_4";
-		//smc0_5v3v_level = <0>;
-		smc0_reset_pin = "GPIOX_17";
-		smc0_reset_level = <0>;
-		smc0_clk_pinmux_reg = <0x2030>;
-		smc0_clk_pinmux_bit = <0x80000>;
-		smc0_clk_oen_reg = <0x200C>;
-		smc0_clk_out_reg = <0x200D>;
-		//smc0_clk_pin = "GPIOX_18";
-		smc0_clk_bit = <0x40000>;
-		smc0_det_pin = "GPIOX_16";
-		smc0_det_invert = <0>;
-		pinctrl-names = "default";
-		pinctrl-0 = <&smc_pins>;
-	};
-
-/// ***************************************************************************************
-///	-	RNG
-//$$ MODULE="RNG"
-//$$ DEVICE="meson-rng"
-//$$ L2 PROP_STR = "status"
-	meson-rng{
-		compatible = "amlogic,meson-rng";
-		status = "ok";
-	};
-
 }; /* end of / */
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
old mode 100755
new mode 100644
index 305d7b5..13a4e65
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_1G_emmc_sdio.dtd
@@ -142,11 +142,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -157,7 +157,7 @@ void root_func(){
 		compatible = "amlogic,deinterlace";
 		dev_name = "deinterlace";
 		status = "ok";
-		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
+		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
 /// ***************************************************************************************
@@ -171,7 +171,7 @@ void root_func(){
 		dev_name = "mesonstream.0";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -188,6 +188,19 @@ void root_func(){
 //		reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+///	-	picture decoder
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";								
+		status = "okay";
+        share-memory-name = "vdec0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x04000000>; //64M
+	}; 
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
@@ -236,16 +249,16 @@ void root_func(){
 //$$ L2 PROP_U32 2 = "reg"
 //$$ L2 PROP_U32 = "irq"
 //$$ L2 PROP_U32 = "vdin_id"
-    vdin1{
-        compatible = "amlogic,vdin";
-        dev_name = "vdin1";
-        status = "ok";
-        reserve-memory = <0x01000000>;
-        miracast_size = <1920 1080>;//1920x1080x2x4=17M
-        reserve-iomap = "true";
-        irq = <117>;
-        vdin_id = <1>;
-    };
+//    vdin1{
+//        compatible = "amlogic,vdin";
+//        dev_name = "vdin1";
+//        status = "ok";
+//        reserve-memory = <0x01000000>;
+//	  miracast_size = <1920 1080>;//1920x1080x2x4=17M
+//	  reserve-iomap = "true";
+//        irq = <117>;
+//        vdin_id = <1>;
+//	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-AMLVIDEO2
@@ -253,13 +266,13 @@ void root_func(){
 //$$ DEVICE="amlvideo2"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2 ="reg"
-    amlvideo2{
-        compatible = "amlogic,amlvideo2";
-        dev_name = "amlvideo2.0";
-        status = "okay";
-        reserve-memory = <0x01800000>;
-        reserve-iomap = "true";
-    };
+//    amlvideo2{
+//		compatible = "amlogic,amlvideo2";
+//		dev_name = "amlvideo2.0";
+//		status = "okay";
+//		reserve-memory = <0x01800000>;
+//		reserve-iomap = "true";
+//	};
 
 /// ***************************************************************************************
 ///	-	ION
@@ -467,7 +480,7 @@ sdio{
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            // lhh gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -945,8 +958,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1440,13 +1453,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-			amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+		
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
index 484ac36..4789bb2 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -457,7 +457,7 @@ sdio{
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            // lhh gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -936,8 +936,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1430,13 +1430,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-			amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+		
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
index 7f1231b..80620bb 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -457,7 +457,7 @@ sdio{
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+        //yxg    gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -934,8 +934,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1428,13 +1428,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-			amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+		
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
old mode 100755
new mode 100644
index e70ba5f..f044e11
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_1G.dtd
@@ -83,14 +83,12 @@ void root_func(){
 		phys_offset = <0x00200000>;
 		linux,total-memory = <0x3fe00000>;
 
-        #address-cells = <1>;
-        #size-cells = <1>;
 //$$ DEVICE="mem_4k2k_decoder"
 //$$ L3 PROP_U32 2 ="reg" 
 		mem_4k2k_decoder: region@1 {
 			region_name = "cma_4k2k";
 			linux,contiguous-region;
-			reg = <0x00000000 0x0ca00000>;
+			reg = <0x00000000 0x0c000000>;
 		};
 
         cma_0:region@0 {
@@ -140,11 +138,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
@@ -155,7 +153,7 @@ void root_func(){
 		compatible = "amlogic,deinterlace";
 		dev_name = "deinterlace";
 		status = "ok";
-		reserve-memory = <0x01800000>; //8x1920x1088x3/2=24M
+		reserve-memory = <0x02100000>; //10x1920x1088x3/2=33M
 	};
 
 /// ***************************************************************************************
@@ -169,7 +167,7 @@ void root_func(){
 		dev_name = "mesonstream.0";
 		status = "okay";
 		reserve-memory = <0x02000000>; // 32M
-		reserve-iomap = "true";
+		//reserve-iomap = "true";
 	};
 
 /// ***************************************************************************************
@@ -186,6 +184,19 @@ void root_func(){
 //		reserve-iomap = "true";
 		linux,contiguous-region = <&mem_4k2k_decoder>;
         };
+///	-	picture decoder
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";								
+		status = "okay";
+        share-memory-name = "vdec0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x04000000>; //64M
+	}; 
 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
@@ -224,24 +235,6 @@ void root_func(){
 //		dev_name = "amvdec_656in";
 //		status = "ok";
 //		reg = <0x1e400000 0x00100000>;
-/// ***************************************************************************************
-///	-	DISP&MM-VDIN
-//$$ MODULE = "DISP&MM-VDIN"
-//$$ DEVICE="vdin0"
-//$$ L2 PROP_STR = "status"
-//$$ L2 PROP_U32 2 = "reg"
-//$$ L2 PROP_U32 = "irq"
-//$$ L2 PROP_U32 = "vdin_id"
-    vdin0{
-        compatible = "amlogic,vdin";
-        dev_name = "vdin0";
-        status = "ok";
-	reserve-memory = <0x00100000>;//only for mpeg2vdin use
-	 reserve-iomap = "true";
-        irq = <115>;
-        vdin_id = <0>;
-	};
-	
 //	};
 //
 /// ***************************************************************************************
@@ -252,16 +245,16 @@ void root_func(){
 //$$ L2 PROP_U32 2 = "reg"
 //$$ L2 PROP_U32 = "irq"
 //$$ L2 PROP_U32 = "vdin_id"
-    vdin1{
-        compatible = "amlogic,vdin";
-        dev_name = "vdin1";
-        status = "ok";
-        reserve-memory = <0x01000000>;
-        miracast_size = <1920 1080>;//1920x1080x2x4=17M
-        reserve-iomap = "true";
-        irq = <117>;
-        vdin_id = <1>;
-    };
+//    vdin1{
+//        compatible = "amlogic,vdin";
+//        dev_name = "vdin1";
+//        status = "ok";
+//        reserve-memory = <0x01000000>;
+//	  miracast_size = <1920 1080>;//1920x1080x2x4=17M
+//	  reserve-iomap = "true";
+//        irq = <117>;
+//        vdin_id = <1>;
+//	};
 
 /// ***************************************************************************************
 ///	-	DISP&MM-AMLVIDEO2
@@ -269,13 +262,13 @@ void root_func(){
 //$$ DEVICE="amlvideo2"
 //$$ L2 PROP_STR = "status"
 //$$ L3 PROP_U32 2 ="reg"
-    amlvideo2{
-        compatible = "amlogic,amlvideo2";
-        dev_name = "amlvideo2.0";
-        status = "okay";
-        reserve-memory = <0x01800000>;
-        reserve-iomap = "true";
-    };
+//    amlvideo2{
+//		compatible = "amlogic,amlvideo2";
+//		dev_name = "amlvideo2.0";
+//		status = "okay";
+//		reserve-memory = <0x01800000>;
+//		reserve-iomap = "true";
+//	};
 
 /// ***************************************************************************************
 ///	-	ION
@@ -463,7 +456,6 @@ sdio{
 //$$ L2 PROP_STR 3 = "caps"
 //$$ L3 PROP_U32 = "f_min"
 //$$ L3 PROP_U32 = "f_max"
-//$$ L3 PROP_U32 = "f_max_w"
 //$$ L3 PROP_U32 = "max_req_size"
 //$$ L2 PROP_STR = "gpio_dat3"
 //$$ L2 PROP_STR = "jtag_pin"
@@ -480,12 +472,11 @@ sdio{
             caps = "MMC_CAP_4_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED";
             f_min = <300000>;
             f_max = <50000000>;
-            f_max_w = <50000000>;
             max_req_size = <0x20000>;          /**128KB*/
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            // lhh gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -541,20 +532,6 @@ sdio{
         };
     };
 
-/// ***************************************************************************************
-///	-	MMC
-//$$ MODULE="MMC"
-//$$ DEVICE="aml_sdhc"
-//$$ L2 PROP_STR = "status"
-//$$ L3 PROP_U32 2 ="reg"
-//$$ L2 PROP_STR 6 = "pinctrl-names"
-//$$ L2 PROP_CHOICE "sdhc_pin_0_match" = "pinctrl-0"
-//$$ L2 PROP_CHOICE "sdhc_pin_1_match" = "pinctrl-1"
-//$$ L2 PROP_CHOICE "sdhc_pin_2_match" = "pinctrl-2"
-//$$ L2 PROP_CHOICE "sdhc_pin_3_match" = "pinctrl-3"
-//$$ L2 PROP_CHOICE "sdhc_pin_4_match" = "pinctrl-4"
-//$$ L2 PROP_CHOICE "sdjc_pin_5_match" = "pinctrl-5"
-// L2 PROP_CHOICE "sdhc_pin_6_match" = "pinctrl-6"
     sdhc{
         compatible = "amlogic,aml_sdhc";
         dev_name = "aml_sdhc.0";
@@ -624,8 +601,8 @@ sdio{
             ocr_avail = <0x00200080>; // 3.3:0x200000, 1.8+3.3:0x00200080
             caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_ERASE", "MMC_CAP_HW_RESET"; // MMC_CAP_NEEDS_POLL -- for detect, MMC_CAP_NONREMOVABLE -- for eMMC/TSD
             caps2 = "MMC_CAP2_HS200_1_8V_SDR";
-	    f_min = <300000>;
-            f_max = <100000000>;
+	        f_min = <300000>;
+            f_max = <150000000>;
             max_req_size = <0x20000>;          /**128KB*/
             gpio_dat3 = "BOOT_3";
             card_type = <1>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -982,8 +959,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1477,13 +1454,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-	    		amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
@@ -1656,13 +1628,6 @@ sdio{
             amlogic,pullupen=<1>;
         };
 
-//$$ MATCH "sdhc_pin_0_match" = "&sdhc_sd_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 4*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sd_clk_cmd_pins:sdhc_sd_clk_cmd_pins{
             amlogic,setmask=<2 0x00000030>;         /*sdhc b*/
             amlogic,clrmask=<5 0x00007c00           /*sdhc a*/
@@ -1674,13 +1639,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_1_match" = "&sdhc_sd_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 4*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 6 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sd_all_pins:sdhc_sd_all_pins{
             amlogic,setmask=<2 0x000000f0>;         /*sdhc b*/
             amlogic,clrmask=<5 0x00007c00           /*sdhc a*/
@@ -1692,13 +1650,7 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_2_match" = "&sdhc_emmc_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
+
         sdhc_emmc_clk_cmd_pins:sdhc_emmc_clk_cmd_pins{
             amlogic,setmask=<4 0x0c000000>;         /*bit[26-27] */
             amlogic,clrmask=<2 0x04c000f0           /*sdhc b & nand*/
@@ -1709,13 +1661,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_3_match" = "&sdhc_emmc_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 10 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_emmc_all_pins:sdhc_emmc_all_pins{
             amlogic,setmask=<4 0x7c000000>;         /*sdhc c*/
             amlogic,clrmask=<2 0x04c000f0           /*sdhc b & nand*/
@@ -1726,13 +1671,7 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_4_match" = "&sdhc_sdio_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
+
         sdhc_sdio_clk_cmd_pins:sdhc_sdio_clk_cmd_pins{
             amlogic,setmask=<5 0x00000c00>;         /*sdhc a bit[10-11] */
             amlogic,clrmask=<2 0x058000f0           /*sdhc b*/
@@ -1743,13 +1682,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_5_match" = "&sdhc_sdio_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 6 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sdio_all_pins:sdhc_sdio_all_pins{
             amlogic,setmask=<5 0x00006c00>;         /*sdhc a*/
             amlogic,clrmask=<2 0x058000f0           /*sdhc b*/
@@ -1911,7 +1843,7 @@ sdio{
                 phy_interface = <1>;
                 new_maclogic = <1>;
                 reset_pin_enable = <0>;
-                reset_pin = "GPIOH_4";
+                reset_pin = "GPI0H_4";
                 reset_delay = <100>;
 	};
 
diff --git a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
old mode 100755
new mode 100644
index 3c6f729..8de8102
--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 
@@ -455,7 +455,6 @@ sdio{
 //$$ L2 PROP_STR 3 = "caps"
 //$$ L3 PROP_U32 = "f_min"
 //$$ L3 PROP_U32 = "f_max"
-//$$ L3 PROP_U32 = "f_max_w"
 //$$ L3 PROP_U32 = "max_req_size"
 //$$ L2 PROP_STR = "gpio_dat3"
 //$$ L2 PROP_STR = "jtag_pin"
@@ -472,12 +471,11 @@ sdio{
             caps = "MMC_CAP_4_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED";
             f_min = <300000>;
             f_max = <50000000>;
-            f_max_w = <50000000>;
             max_req_size = <0x20000>;          /**128KB*/
             gpio_dat3 = "CARD_4";
             jtag_pin = "CARD_0";
             gpio_cd = "CARD_6";
-            gpio_ro = "GPIOZ_0";
+            //gpio_ro = "GPIOZ_0";
             irq_in = <3>;
             irq_out = <5>;
             card_type = <5>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -533,20 +531,6 @@ sdio{
         };
     };
 
-/// ***************************************************************************************
-///	-	MMC
-//$$ MODULE="MMC"
-//$$ DEVICE="aml_sdhc"
-//$$ L2 PROP_STR = "status"
-//$$ L3 PROP_U32 2 ="reg"
-//$$ L2 PROP_STR 6 = "pinctrl-names"
-//$$ L2 PROP_CHOICE "sdhc_pin_0_match" = "pinctrl-0"
-//$$ L2 PROP_CHOICE "sdhc_pin_1_match" = "pinctrl-1"
-//$$ L2 PROP_CHOICE "sdhc_pin_2_match" = "pinctrl-2"
-//$$ L2 PROP_CHOICE "sdhc_pin_3_match" = "pinctrl-3"
-//$$ L2 PROP_CHOICE "sdhc_pin_4_match" = "pinctrl-4"
-//$$ L2 PROP_CHOICE "sdjc_pin_5_match" = "pinctrl-5"
-// L2 PROP_CHOICE "sdhc_pin_6_match" = "pinctrl-6"
     sdhc{
         compatible = "amlogic,aml_sdhc";
         dev_name = "aml_sdhc.0";
@@ -616,8 +600,8 @@ sdio{
             ocr_avail = <0x00200080>; // 3.3:0x200000, 1.8+3.3:0x00200080
             caps = "MMC_CAP_8_BIT_DATA","MMC_CAP_MMC_HIGHSPEED","MMC_CAP_SD_HIGHSPEED", "MMC_CAP_NONREMOVABLE","MMC_CAP_ERASE", "MMC_CAP_HW_RESET"; // MMC_CAP_NEEDS_POLL -- for detect, MMC_CAP_NONREMOVABLE -- for eMMC/TSD
             caps2 = "MMC_CAP2_HS200_1_8V_SDR";
-	    f_min = <300000>;
-            f_max = <100000000>;
+	        f_min = <300000>;
+            f_max = <150000000>;
             max_req_size = <0x20000>;          /**128KB*/
             gpio_dat3 = "BOOT_3";
             card_type = <1>; /* 0:unknown, 1:mmc card(include eMMC), 2:sd card(include tSD), 3:sdio device(ie:sdio-wifi), 4:SD combo (IO+mem) card, 5:NON sdio device(means sd/mmc card), other:reserved */
@@ -974,8 +958,8 @@ sdio{
 		status = "ok";
 		reg = <0xd0048600 0x24>;
 		pinctrl-names = "nand_rb_mod","nand_norb_mod";
-		pinctrl-0 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
-		pinctrl-1 = <&nand_input_state &conf_nand_state &conf_nand_state1 &nand_base>;
+		pinctrl-0 = <&nand_input_state &conf_nand_state &nand_base>;
+		pinctrl-1 = <&nand_input_state &conf_nand_state &nand_base>;
 		device_id = <0>;
 		plat-names = "nandnormal";
 		plat-num = <1>;
@@ -1469,13 +1453,8 @@ sdio{
 			amlogic,pins = "BOOT_0","BOOT_1","BOOT_2","BOOT_3","BOOT_4",
 						"BOOT_5","BOOT_6","BOOT_7","BOOT_15";
 			amlogic,pullup=<0>;
-	    		amlogic,pullupen=<1>;
-		};
-		conf_nand_state1: conf_nand1{
-			amlogic,pins = "BOOT_8", "BOOT_10";
-			amlogic,pullup=<1>;
-			amlogic,pullupen=<1>;
 		};
+
 //$$ MATCH "Nand_pin_0_match" = "&nand_base"
 //$$ MATCH "Nand_pin_1_match" = "&nand_base"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
@@ -1648,13 +1627,6 @@ sdio{
             amlogic,pullupen=<1>;
         };
 
-//$$ MATCH "sdhc_pin_0_match" = "&sdhc_sd_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 4*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sd_clk_cmd_pins:sdhc_sd_clk_cmd_pins{
             amlogic,setmask=<2 0x00000030>;         /*sdhc b*/
             amlogic,clrmask=<5 0x00007c00           /*sdhc a*/
@@ -1666,13 +1638,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_1_match" = "&sdhc_sd_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 4*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 6 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sd_all_pins:sdhc_sd_all_pins{
             amlogic,setmask=<2 0x000000f0>;         /*sdhc b*/
             amlogic,clrmask=<5 0x00007c00           /*sdhc a*/
@@ -1684,13 +1649,7 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_2_match" = "&sdhc_emmc_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
+
         sdhc_emmc_clk_cmd_pins:sdhc_emmc_clk_cmd_pins{
             amlogic,setmask=<4 0x0c000000>;         /*bit[26-27] */
             amlogic,clrmask=<2 0x04c000f0           /*sdhc b & nand*/
@@ -1701,13 +1660,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_3_match" = "&sdhc_emmc_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 10 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_emmc_all_pins:sdhc_emmc_all_pins{
             amlogic,setmask=<4 0x7c000000>;         /*sdhc c*/
             amlogic,clrmask=<2 0x04c000f0           /*sdhc b & nand*/
@@ -1718,13 +1670,7 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_4_match" = "&sdhc_sdio_clk_cmd_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 2 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
+
         sdhc_sdio_clk_cmd_pins:sdhc_sdio_clk_cmd_pins{
             amlogic,setmask=<5 0x00000c00>;         /*sdhc a bit[10-11] */
             amlogic,clrmask=<2 0x058000f0           /*sdhc b*/
@@ -1735,13 +1681,6 @@ sdio{
             amlogic,pullup=<1>;
             amlogic,pullupen=<1>;
         };
-//$$ MATCH "sdhc_pin_5_match" = "&sdhc_sdio_all_pins"
-//$$ L2 PROP_U32 2 = "amlogic,setmask"
-//$$ L2 PROP_U32 3*2 = "amlogic,clrmask"
-//$$ L2 PROP_STR 6 = "amlogic,pins"
-//$$ L2 PROP_U32 = "amlogic,enable-output"
-//$$ L2 PROP_U32 = "amlogic,pullup"
-//$$ L2 PROP_U32 = "amlogic,pullupen"
         sdhc_sdio_all_pins:sdhc_sdio_all_pins{
             amlogic,setmask=<5 0x00006c00>;         /*sdhc a*/
             amlogic,clrmask=<2 0x058000f0           /*sdhc b*/
@@ -1903,7 +1842,7 @@ sdio{
                 phy_interface = <1>;
                 new_maclogic = <1>;
                 reset_pin_enable = <0>;
-                reset_pin = "GPIOH_4";
+                reset_pin = "GPI0H_4";
                 reset_delay = <100>;
 	};
 
@@ -1954,14 +1893,4 @@ sdio{
 		pinctrl-0 = <&dvb_fe_pins>;
 	};
 
-/// ***************************************************************************************
-///	-	RNG
-//$$ MODULE="RNG"
-//$$ DEVICE="meson-rng"
-//$$ L2 PROP_STR = "status"
-	meson-rng{
-		compatible = "amlogic,meson-rng";
-		status = "ok";
-	};
-
 }; /* end of / */
