diff -Naur newt-0.52.19/Makefile.in newt-0.52.19.patch/Makefile.in
--- newt-0.52.19/Makefile.in	2016-03-23 16:46:24.000000000 +0100
+++ newt-0.52.19.patch/Makefile.in	2016-10-23 10:30:52.742825943 +0200
@@ -102,7 +102,7 @@
 	done || :
 	touch $@
 
-whiptail: $(NDIALOGOBJS) $(LIBNEWTSH)
+whiptail: $(NDIALOGOBJS) $(LIBNEWT)
 	$(CC) -g -o whiptail $(NDIALOGOBJS) -L. $(LDFLAGS) -lnewt $(LIBS) -lpopt
 
 whiptcl.$(SOEXT): $(WHIPTCLOBJS) $(LIBNEWTSH)
