--- a/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd	2015-04-29 16:57:27.361145046 +0300
+++ b/arch/arm/boot/dts/amlogic/meson8m2_n200_2G.dtd	2015-04-29 16:57:32.989130612 +0300
@@ -1895,7 +1895,7 @@
 /// ***************************************************************************************
 ///	-	Ethernet
         ethmac: ethernet@0xc9410000{
-                compatible = "amlogic,meson8m2-rmii-dwmac", "snps,dwmac";
+                compatible = "amlogic,meson8m2-rgmii-dwmac", "snps,dwmac";
                 reg = <0xc9410000 0x10000
                        0xc1108108 0x4>;
                 interrupts = <0 8 1>;
