diff -Nur a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp c/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp
--- a/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2015-11-12 02:09:30.000000000 +0100
+++ c/xbmc/cores/AudioEngine/Sinks/AESinkALSA.cpp	2015-11-17 20:02:49.783734486 +0100
@@ -1173,7 +1173,8 @@
        * "plughw", "dsnoop"). */
 
       else if (baseName != "default"
-            && baseName != "sysdefault"
+          //FIXME: Until better solution is found, whitelist sysdefault devices
+          //  && baseName != "sysdefault"
             && baseName != "surround40"
             && baseName != "surround41"
             && baseName != "surround50"
@@ -1288,6 +1289,10 @@
     return AE_DEVTYPE_HDMI;
   else if (name.substr(0, 6) == "iec958" || name.substr(0, 5) == "spdif")
     return AE_DEVTYPE_IEC958;
+  
+  //FIXME: hack for sunxi hdmi
+  if(name.find("sndhdmi") != std::string::npos)
+    return AE_DEVTYPE_HDMI;
 
   return AE_DEVTYPE_PCM;
 }
