diff -Naur dtc-0931cea/Makefile dtc-0931cea.patch/Makefile
--- dtc-0931cea/Makefile	2016-12-26 23:44:17.000000000 +0100
+++ dtc-0931cea.patch/Makefile	2017-01-01 15:50:41.378990810 +0100
@@ -18,7 +18,7 @@
 CPPFLAGS = -I libfdt -I .
 WARNINGS = -Wall -Wpointer-arith -Wcast-qual -Wnested-externs \
 	-Wstrict-prototypes -Wmissing-prototypes -Wredundant-decls -Wshadow
-CFLAGS = -g -Os -fPIC -Werror $(WARNINGS)
+CFLAGS = -g -Os -fPIC $(WARNINGS)
 
 BISON = bison
 LEX = flex
