diff -Naur a/Makefile.am b/Makefile.am
--- a/Makefile.am	2015-11-25 08:00:27.000000000 +0000
+++ b/Makefile.am	2016-03-01 19:30:08.475070195 +0000
@@ -166,6 +166,9 @@
 	src/libbluray/keys.h \
 	src/libbluray/player_settings.h \
 	src/libbluray/bdnav/clpi_data.h \
+	src/libbluray/bdnav/clpi_parse.h \
+	src/libbluray/bdnav/mpls_parse.h \
+	src/libbluray/bdnav/uo_mask_table.h \
 	src/libbluray/bdnav/meta_data.h \
 	src/libbluray/decoders/overlay.h \
 	src/util/log_control.h
