diff --git a/menu/drivers_display/menu_display_gl.c b/menu/drivers_display/menu_display_gl.c
index c4424e8..cf70703 100644
--- a/menu/drivers_display/menu_display_gl.c
+++ b/menu/drivers_display/menu_display_gl.c
@@ -182,6 +182,8 @@ static void menu_display_gl_draw_pipeline(void *data)
             video_shader_driver_use(&shader_info);
 
             t += 0.01;
+            if (t > 50.0)
+               t -= 50.0;
 
             uniform_param.enabled           = true;
             uniform_param.lookup.enable     = true;
