diff --git a/kernel/nv-mmap.c b/kernel/nv-mmap.c
index 178c6b7..b969bfd 100644
--- a/kernel/nv-mmap.c
+++ b/kernel/nv-mmap.c
@@ -463,7 +463,11 @@ int nv_kern_mmap(
         NV_PRINT_AT(NV_DBG_MEMINFO, at);
         nv_vm_list_page_count(&at->page_table[i], pages);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,7,0)
+        vma->vm_flags |= (VM_IO | VM_LOCKED | (VM_DONTEXPAND | VM_DONTDUMP));
+#else
         vma->vm_flags |= (VM_IO | VM_LOCKED | VM_RESERVED);
+#endif
         vma->vm_flags |= (VM_DONTEXPAND | VM_DONTDUMP);
     }
 

