diff -Nur a/sound/soc/sunxi/audiocodec/sun8iw7_sndcodec.c b/sound/soc/sunxi/audiocodec/sun8iw7_sndcodec.c
--- a/sound/soc/sunxi/audiocodec/sun8iw7_sndcodec.c	2016-04-24 18:52:22.000000000 +0200
+++ b/sound/soc/sunxi/audiocodec/sun8iw7_sndcodec.c	2016-05-12 18:28:34.295125768 +0200
@@ -627,6 +627,10 @@
 	/*enable dac digital*/
 	codec_wr_control(SUNXI_DAC_DPC, 0x1, DAC_EN, 0x1);
 	codec_wr_prcm_control(LINEOUT_PA_GAT, 0x1, PA_CLK_GC, 0x0);
+	
+	/* This is workaround for loud poping anytime sound is opened */
+	codec_wr_prcm_control(ROMIXSC, 0x1, RMIXMUTEDACR, 0);
+	codec_wr_prcm_control(LOMIXSC, 0x1, LMIXMUTEDACL, 0);
 
 	/*set TX FIFO send drq level*/
 	codec_wr_control(SUNXI_DAC_FIFOC ,0x7f, TX_TRI_LEVEL, 0xf);
