diff -Naur libpng-1.6.26/pngrutil.c libpng-1.6.26.patch/pngrutil.c
--- libpng-1.6.26/pngrutil.c	2016-10-20 02:07:24.000000000 +0200
+++ libpng-1.6.26.patch/pngrutil.c	2016-10-30 13:58:23.305058239 +0100
@@ -418,7 +418,7 @@
             png_ptr->flags |= PNG_FLAG_ZSTREAM_INITIALIZED;
       }
 
-#if ZLIB_VERNUM >= 0x1281
+#if ZLIB_VERNUM >= 0x1281 && ZLIB_VERNUM != 0x128f
       /* Turn off validation of the ADLER32 checksum */
       if ((png_ptr->flags & PNG_FLAG_CRC_CRITICAL_IGNORE) != 0)
          ret = inflateValidate(&png_ptr->zstream, 0);
