--- a/src/include/os/rt_linux.h	2014-09-18 18:04:41.000000000 +0400
+++ b/src/include/os/rt_linux.h	2014-09-21 06:27:26.741742567 +0400
@@ -127,7 +127,7 @@
 #ifdef CONFIG_STA_SUPPORT
 
 #ifdef RTMP_MAC_USB
-#define STA_PROFILE_PATH			"/etc/Wireless/RT2870STA/RT2870STA.dat"
+#define STA_PROFILE_PATH			"/lib/firmware/MT7601USTA/MT7601USTA.dat"
 #define STA_DRIVER_VERSION			"3.0.0.3"
 #ifdef MULTIPLE_CARD_SUPPORT
 #define CARD_INFO_PATH			"/etc/Wireless/RT2870STA/RT2870STACard.dat"
