diff -Naur kodi-eceb0d6/system/keymaps/keyboard.xml kodi-eceb0d6.patch/system/keymaps/keyboard.xml
--- kodi-eceb0d6/system/keymaps/keyboard.xml	2016-03-29 21:36:04.000000000 +0200
+++ kodi-eceb0d6.patch/system/keymaps/keyboard.xml	2016-03-30 09:49:49.388585853 +0200
@@ -95,6 +95,7 @@
       <sleep>ActivateWindow(shutdownmenu)</sleep>
       <!-- PVR windows -->
       <e>ActivateWindow(TVGuide)</e>
+      <epg>ActivateWindow(TVGuide)</epg>
       <h>ActivateWindow(TVChannels)</h>
       <j>ActivateWindow(RadioChannels)</j>
       <k>ActivateWindow(TVRecordings)</k>
@@ -221,6 +222,7 @@
     <keyboard>
       <e>PreviousMenu</e>
       <t>ShowTimerRule</t>
+      <epg>PreviousMenu</epg>
     </keyboard>
   </TVGuide>
   <MyRadioChannels>
diff -Naur kodi-eceb0d6/xbmc/input/linux/LinuxInputDevices.cpp kodi-eceb0d6.patch/xbmc/input/linux/LinuxInputDevices.cpp
--- kodi-eceb0d6/xbmc/input/linux/LinuxInputDevices.cpp	2016-03-29 21:36:04.000000000 +0200
+++ kodi-eceb0d6.patch/xbmc/input/linux/LinuxInputDevices.cpp	2016-03-30 09:48:10.182063910 +0200
@@ -273,6 +273,7 @@
   { KEY_FILE          , XBMCK_LAUNCH_FILE_BROWSER},
   { KEY_SELECT        , XBMCK_RETURN      },
   { KEY_CONFIG        , XBMCK_CONFIG      },
+  { KEY_EPG           , XBMCK_EPG         },
   // The Little Black Box Remote Additions
   { 384               , XBMCK_LEFT        }, // Red
   { 378               , XBMCK_RIGHT       }, // Green
diff -Naur kodi-eceb0d6/xbmc/input/XBMC_keysym.h kodi-eceb0d6.patch/xbmc/input/XBMC_keysym.h
--- kodi-eceb0d6/xbmc/input/XBMC_keysym.h	2016-03-29 21:36:04.000000000 +0200
+++ kodi-eceb0d6.patch/xbmc/input/XBMC_keysym.h	2016-03-30 09:48:10.183063895 +0200
@@ -229,6 +229,7 @@
   XBMCK_FAVORITES   = 0x14d,
   XBMCK_HOMEPAGE    = 0x14e,
   XBMCK_CONFIG      = 0x14f,
+  XBMCK_EPG         = 0x150,
 
   // Add any other keys here
 
diff -Naur kodi-eceb0d6/xbmc/input/XBMC_keytable.cpp kodi-eceb0d6.patch/xbmc/input/XBMC_keytable.cpp
--- kodi-eceb0d6/xbmc/input/XBMC_keytable.cpp	2016-03-29 21:36:04.000000000 +0200
+++ kodi-eceb0d6.patch/xbmc/input/XBMC_keytable.cpp	2016-03-30 09:48:10.183063895 +0200
@@ -243,6 +243,7 @@
 , { XBMCK_FAVORITES,              0,    0, XBMCVK_FAVORITES,     "favorites" }
 , { XBMCK_HOMEPAGE ,              0,    0, XBMCVK_HOMEPAGE,      "homepage" }
 , { XBMCK_CONFIG,                 0,    0, XBMCVK_CONFIG,        "config" }
+, { XBMCK_EPG   ,                 0,    0, XBMCVK_EPG,           "epg" }
 };
 
 static int XBMCKeyTableSize = sizeof(XBMCKeyTable)/sizeof(XBMCKEYTABLE);
diff -Naur kodi-eceb0d6/xbmc/input/XBMC_vkeys.h kodi-eceb0d6.patch/xbmc/input/XBMC_vkeys.h
--- kodi-eceb0d6/xbmc/input/XBMC_vkeys.h	2016-03-29 21:36:04.000000000 +0200
+++ kodi-eceb0d6.patch/xbmc/input/XBMC_vkeys.h	2016-03-30 09:48:10.184063881 +0200
@@ -221,6 +221,7 @@
   XBMCVK_FAVORITES      = 0xE9,
   XBMCVK_HOMEPAGE       = 0xEA,
   XBMCVK_CONFIG         = 0xEB,
+  XBMCVK_EPG            = 0xEC,
 
   XBMCVK_LAST           = 0xFF
 } XBMCVKey;
