diff -Naur kodi-c6a62cf/project/cmake/modules/FindPython.cmake kodi-c6a62cf.patch/project/cmake/modules/FindPython.cmake
--- kodi-c6a62cf/project/cmake/modules/FindPython.cmake	2016-09-17 15:57:58.000000000 +0200
+++ kodi-c6a62cf.patch/project/cmake/modules/FindPython.cmake	2016-09-23 19:57:34.036971538 +0200
@@ -16,15 +16,11 @@
     find_path(PYTHON_INCLUDE_DIRS NAMES Python.h PATHS ${DEPENDS_PATH}/include/python2.6 ${DEPENDS_PATH}/include/python2.7)
     set(PYTHON_INCLUDE_DIR ${PYTHON_INCLUDE_DIRS} CACHE PATH "python include dir" FORCE)
 
-    find_library(FFI_LIBRARY ffi)
-    find_library(EXPAT_LIBRARY expat)
-    find_library(INTL_LIBRARY intl)
-
     if(NOT CORE_SYSTEM_NAME STREQUAL android)
         set(PYTHON_DEP_LIBRARIES -lpthread -ldl -lutil)
     endif()
 
-    set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${FFI_LIBRARY} ${EXPAT_LIBRARY} ${INTL_LIBRARY} ${PYTHON_DEP_LIBRARIES}
+    set(PYTHON_LIBRARIES ${PYTHON_LIBRARY} ${PYTHON_DEP_LIBRARIES}
         CACHE INTERNAL "python libraries" FORCE)
   else()
     find_package(PythonLibs 2.7 REQUIRED)
