diff -Naur makemkv-oss-1.8.5/makefile.linux makemkv-oss-1.8.5.patch/makefile.linux
--- makemkv-oss-1.8.5/makefile.linux	2013-09-13 02:30:37.000000000 +0200
+++ makemkv-oss-1.8.5.patch/makefile.linux	2013-09-22 16:38:19.096662564 +0200
@@ -1,7 +1,7 @@
 
 include makefile.common
 
-GCC=gcc
+GCC:=gcc
 MOC=$(strip $(shell which moc-qt4))
 ifeq ($(MOC),)
 MOC=moc
@@ -38,7 +38,7 @@
 	install -D -m 644 makemkvgui/share/icons/128x128/makemkv.png $(DESTDIR)$(PREFIX)/share/icons/hicolor/128x128/apps/makemkv.png
 
 out/%: out/%.full
-	objcopy --strip-all --strip-debug --strip-unneeded --discard-all $< $@ 
+	$(OBJCOPY) --strip-all --strip-debug --strip-unneeded --discard-all $< $@ 
 
 out/libdriveio.so.0.full:
 	mkdir -p out
@@ -83,5 +83,5 @@
 tmp/image_data.o : makemkvgui/bin/image_data.bin
 	mkdir -p tmp
 	ld -r -b binary -o $@.rw $<
-	objcopy --rename-section .data=.rodata2,readonly,alloc,load,data,contents $@.rw $@
+	$(OBJCOPY) --rename-section .data=.rodata2,readonly,alloc,load,data,contents $@.rw $@
 
