--- a/include/configs/sun8iw7p1.h	2015-01-27 03:29:01.000000000 +0100
+++ b/include/configs/sun8iw7p1.h	2015-11-10 20:39:51.824642276 +0100
@@ -318,7 +318,14 @@
 
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	"bootdelay=3\0" \
-	"bootcmd=run setargs_nand boot_normal\0" \
+	"bootcmd=run loaduenv setargs_nand boot_normal\0" \
+	"scriptaddr=0x44000000\0" \
+	"loadbootenv=" \
+		"fatload mmc 0:1 $scriptaddr uEnv.txt\0" \
+	"loaduenv=if run loadbootenv; then " \
+		"env import -t ${scriptaddr} ${filesize};" \
+		"run uenvcmd;" \
+		"fi\0" \
 	"console=ttyS0,115200\0" \
 	"nand_root=/dev/nandd\0" \
 	"mmc_root=/dev/mmcblk0p7\0" \
