Removed some undefined declarations

--- a/include/axp_power.h	2015-01-27 03:29:01.000000000 +0100
+++ b/include/axp_power.h	2015-11-03 22:17:50.164093110 +0100
@@ -207,6 +207,4 @@
 extern int axp_power_init(void);
 extern int axp_power_ioctl(int cmd,void *arg);
 extern int axp_power_exit(void);
-extern inline int pm_bus_read(unsigned  char  chip, unsigned char addr, unsigned char* data);
-extern inline int pm_bus_write(unsigned char  chip, unsigned char addr, unsigned char data);
 #endif
