diff -Naur retroarch.git/gfx/drivers_context/mali_fbdev_ctx.c retroarch.patch/gfx/drivers_context/mali_fbdev_ctx.c
--- retroarch.git/gfx/drivers_context/mali_fbdev_ctx.c	2015-11-20 14:53:22.904541008 +0100
+++ retroarch.patch/gfx/drivers_context/mali_fbdev_ctx.c	2015-11-20 15:36:40.173833445 +0100
@@ -30,7 +30,7 @@
 #include "../common/egl_common.h"
 #include "../common/gl_common.h"
 
-struct fbdev_window native_window;
+struct mali_native_window native_window;
 static bool g_resize;
 static unsigned g_width, g_height;
 
