diff -Naur linux-3.11.1/arch/x86/boot/Makefile linux-3.11.1.patch/arch/x86/boot/Makefile
--- linux-3.11.1/arch/x86/boot/Makefile	2013-09-14 16:07:01.000000000 +0200
+++ linux-3.11.1.patch/arch/x86/boot/Makefile	2013-10-09 13:27:47.563875692 +0200
@@ -14,7 +14,7 @@
 # Set it to -DSVGA_MODE=NORMAL_VGA if you just want the EGA/VGA mode.
 # The number is the same as you would ordinarily press at bootup.
 
-SVGA_MODE	:= -DSVGA_MODE=NORMAL_VGA
+SVGA_MODE	:= -DSVGA_MODE=792
 
 targets		:= vmlinux.bin setup.bin setup.elf bzImage
 targets		+= fdimage fdimage144 fdimage288 image.iso mtools.conf
