--- a/sound/pci/hda/hda_intel.c.orig	2012-06-21 01:23:34.000000000 +0200
+++ a/sound/pci/hda/hda_intel.c	2012-06-21 07:31:38.636405457 +0200
@@ -94,7 +94,7 @@ MODULE_PARM_DESC(probe_only, "Only probi
 module_param(single_cmd, bool, 0444);
 MODULE_PARM_DESC(single_cmd, "Use single command to communicate with codecs "
 		 "(for debugging only).");
-module_param(enable_msi, bint, 0444);
+module_param(enable_msi, int, 0444);
 MODULE_PARM_DESC(enable_msi, "Enable Message Signaled Interrupt (MSI)");
 #ifdef CONFIG_SND_HDA_PATCH_LOADER
 module_param_array(patch, charp, NULL, 0444);
@@ -122,7 +122,7 @@ MODULE_PARM_DESC(power_save_controller,
 #endif
 
 static int align_buffer_size = -1;
-module_param(align_buffer_size, bint, 0644);
+module_param(align_buffer_size, int, 0644);
 MODULE_PARM_DESC(align_buffer_size,
 		"Force buffer and period sizes to be multiple of 128 bytes.");
 
