diff -Naur linux-4.4.39/net/sunrpc/rpcb_clnt.c linux-4.4.39.patch/net/sunrpc/rpcb_clnt.c
--- linux-4.4.39/net/sunrpc/rpcb_clnt.c	2016-12-15 17:49:34.000000000 +0100
+++ linux-4.4.39.patch/net/sunrpc/rpcb_clnt.c	2016-12-24 23:53:56.262699783 +0100
@@ -36,7 +36,7 @@
 # define RPCDBG_FACILITY	RPCDBG_BIND
 #endif
 
-#define RPCBIND_SOCK_PATHNAME	"/var/run/rpcbind.sock"
+#define RPCBIND_SOCK_PATHNAME	"/run/rpcbind.sock"
 
 #define RPCBIND_PROGRAM		(100000u)
 #define RPCBIND_PORT		(111u)
