diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
index 484ac36..8cfb770 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdhc.dtd
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -179,6 +179,19 @@ void root_func(){
                 reserve-iomap = "true";
         };
 
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";								
+		status = "okay";
+        share-memory-name = "vdec0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x04000000>; //64M
+	}; 
+		
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
@@ -262,7 +275,7 @@ void root_func(){
         status = "ok";
         share-memory-name = "ppmgr0";
         share-memory-offset = <0>;
-        share-memory-size = <0x01000000>; //16M
+        share-memory-size = <0x02000000>; //32M
     };
 /// ***************************************************************************************
 ///	-	DISP&MM-Vout
@@ -296,7 +309,7 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
     rtc{
 		compatible = "amlogic,aml_rtc";
-		status = "okay";
+		status = "disabled";
 	};
 
 /// ***************************************************************************************
@@ -630,7 +643,7 @@ sdio{
 		pinctrl-names="default";
 		pinctrl-0=<&ao_i2c_master>;
 		use_pio = <0>;
-		master_i2c_speed = <100000>;
+		master_i2c_speed = <400000>;
 	};
 
 //$$ DEVICE = "I2C_A"
@@ -693,17 +706,17 @@ sdio{
 //$$ L3 PROP_U32 2="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
 //$$ L2 PROP_CHOICE "I2C_D_pin_match" = "pinctrl-0"	
-//	i2c@c1108d20{ /*I2C-D*/
-//		compatible = "amlogic,aml_i2c";
-//		dev_name = "i2c-D";
-//		status = "ok";
-//		reg = <0xc1108d20 0x20>;
-//		device_id = <4>;
-//		pinctrl-names="default";
-//		pinctrl-0=<&d_i2c_master>;
-//		#address-cells = <1>;
-//		#size-cells = <0>;
-//	};
+	i2c@c1108d20{ /*I2C-D*/
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-D";
+		status = "ok";
+		reg = <0xc1108d20 0x20>;
+		device_id = <4>;
+		pinctrl-names="default";
+		pinctrl-0=<&d_i2c_master>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+	};
 
 /// ***************************************************************************************
 ///	-	Power
@@ -813,6 +826,13 @@ sdio{
             vddao_voltage = <1200000>;                          /** default voltage for AO, optional     */
             status = "okay";                                    /** status of pmu, only one should be OK */
         };
+		 rtc8563{
+            compatible = "rtc8563";                           /** for driver probe, must have          */
+            i2c_bus = "i2c_bus_ao";                           /** attached i2c bus, must have          */
+            reg = <0x51>;                                     /** device i2c address, must have        */
+            gpio_interrupt = "GPIOX_12";
+            status = "ok";
+        };
     };
 
 /// ***************************************************************************************
@@ -1070,7 +1090,7 @@ sdio{
     audio_m8_audio{
         compatible = "amlogic,m8_audio_codec";
         dev_name = "aml_m8_codec.0";
-        status = "disable";
+        status = "okay";
     };
 
 //$$ DEVICE="dummy_codec"
@@ -1083,10 +1103,10 @@ sdio{
 
     aml_audio_codec{
         compatible = "amlogic,audio_codec";
-          dummy:dummy{
-            codec_name = "dummy_codec";
-            status = "okay";
-          };
+//		dummy:dummy{
+//			codec_name = "dummy_codec";
+//			status = "okay";
+//			};
 //        rt5616:rt5616{
 //            codec_name = "rt5616";
 //            i2c_addr = <0x1B>;
@@ -1112,6 +1132,15 @@ sdio{
 //            capless = <1>;
 //	    status = "disable";
 //        };
+		es8323:es8323{
+           codec_name = "es8323";
+           i2c_addr = <0x10>;
+           i2c_bus = "i2c_bus_d";
+           id_reg = <0x00>;
+           id_val = <0x06>;
+           capless = <0>;
+           status = "okay";
+        };
     };
 
 //$$ DEVICE="aml_m8_sound_card"
@@ -1126,7 +1155,7 @@ sdio{
     aml_m8_sound_card{
         compatible = "sound_card, aml_snd_m8";
         aml,sound_card = "AML-M8AUDIO";
-        aml,codec_dai = "AML-M8","dummy_codec";
+        aml,codec_dai = "AML-M8","es8323-hifi";
         //aml,codec_name = "aml_m8_codec.0","rt5616.4-001b";
         //aml,audio-codec = <&rt5616>;
         aml,audio-routing-rt5616 = 
@@ -1146,6 +1175,11 @@ sdio{
         aml,audio-routing-dummy=
             "Ext Spk","LOUTL",
             "Ext Spk","LOUTR";
+		aml,audio-routing-es8323 = 
+            "Ext Spk","LOUTL",
+            "Ext Spk","LOUTR",
+            "HP","HPOL",
+            "HP","HPOR";
 
         mute_gpio = "GPIO_BSD_EN";
         hp_disable;
@@ -1727,10 +1761,10 @@ sdio{
 //$$ MATCH "I2C_D_pin_match" = "&d_i2c_master"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_STR 2 = "amlogic,pins"
-//		d_i2c_master:d_i2c{
-//			amlogic,setmask=<4 0xc>;
-//			amlogic,pins="GPIOH_7","GPIOH_8";
-//		};
+		d_i2c_master:d_i2c{
+			amlogic,setmask=<4 0xc>;
+			amlogic,pins="GPIOH_7","GPIOH_8";
+		};
 
 //$$ MATCH "remote_pin_0_match" = "&remote_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
@@ -1745,11 +1779,13 @@ sdio{
 //$$ L2 PROP_U32 4 = "amlogic,clrmask"
 //$$ L2 PROP_STR 8 = "amlogic,pins"
         audio_pins:audio_pin{
-            amlogic,setmask=<3 0x78000000      /*bt pcm*/
+            amlogic,setmask=<
+	    		1 0x20
+	    		3 0x78000000      /*bt pcm*/
                              10 0x78000000>;  /*external audio codec*/
             amlogic,clrmask=<4 0x3c000         /*bt pcm*/            
                              5 0x1000>;        /*bt pcm*/
-            amlogic,pins = "GPIOX_4","GPIOX_5","GPIOX_6","GPIOX_7",        /*bt pcm*/
+            amlogic,pins = "GPIOY_5","GPIOX_4","GPIOX_5","GPIOX_6","GPIOX_7",        /*bt pcm*/
                            "GPIOAO_8","GPIOAO_9","GPIOAO_10","GPIOAO_11";  /*external audio codec*/
         };
 
diff --git a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
index 7f1231b..cba1522 100644
--- a/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k200b_2G_emmc_sdio.dtd
@@ -133,11 +133,11 @@ void root_func(){
 		compatible = "amlogic,mesonfb";
 		dev_name = "mesonfb";
 		status = "okay";
-	   	reserve-memory = <0x01800000  0x00100000>;
+	   	reserve-memory = <0x04800000  0x00100000>;
 	   	reserve-iomap = "true";
                 vmode = <3>; /**0:VMODE_720P 1:VMODE_LCD  2:VMODE_LVDS_1080P 3:VMODE_1080P*/
 		scale_mode = <1>; /*0:default 1:new*/
-		4k2k_fb = <0>;
+		4k2k_fb = <1>;
  		display_size_default = <1920 1080 1920 3240 32>; //1920*1080*4*3 = 0x17BB000
 	};
 	
@@ -179,6 +179,18 @@ void root_func(){
                 reserve-iomap = "true";
         };
 
+//$$ MODULE="picture decoder"
+//$$ DEVICE="picdec"
+//$$ L2 PROP_STR = "status"
+//$$ L3 PROP_U32 2 ="reg"
+    picdec{
+		compatible = "amlogic,picdec";
+		dev_name = "picdec";								
+		status = "okay";
+        share-memory-name = "vdec0";
+        share-memory-offset = <0>;
+        share-memory-size = <0x04000000>; //64M
+	}; 
 /// ***************************************************************************************
 ///	-	DISP&MM-PostProcess
 //$$ MODULE="DISP&MM-PostProcess"
@@ -262,7 +274,7 @@ void root_func(){
         status = "ok";
         share-memory-name = "ppmgr0";
         share-memory-offset = <0>;
-        share-memory-size = <0x01000000>; //16M
+        share-memory-size = <0x02000000>; //32M
     };
 /// ***************************************************************************************
 ///	-	DISP&MM-Vout
@@ -296,7 +308,7 @@ void root_func(){
 //$$ L2 PROP_STR = "status"
     rtc{
 		compatible = "amlogic,aml_rtc";
-		status = "okay";
+		status = "disabled";
 	};
 
 /// ***************************************************************************************
@@ -628,7 +640,7 @@ sdio{
 		pinctrl-names="default";
 		pinctrl-0=<&ao_i2c_master>;
 		use_pio = <0>;
-		master_i2c_speed = <100000>;
+		master_i2c_speed = <400000>;
 	};
 
 //$$ DEVICE = "I2C_A"
@@ -691,17 +703,17 @@ sdio{
 //$$ L3 PROP_U32 2="reg"
 //$$ L3 PROP_STR = "pinctrl-names"
 //$$ L2 PROP_CHOICE "I2C_D_pin_match" = "pinctrl-0"	
-//	i2c@c1108d20{ /*I2C-D*/
-//		compatible = "amlogic,aml_i2c";
-//		dev_name = "i2c-D";
-//		status = "ok";
-//		reg = <0xc1108d20 0x20>;
-//		device_id = <4>;
-//		pinctrl-names="default";
-//		pinctrl-0=<&d_i2c_master>;
-//		#address-cells = <1>;
-//		#size-cells = <0>;
-//	};
+	i2c@c1108d20{ /*I2C-D*/
+		compatible = "amlogic,aml_i2c";
+		dev_name = "i2c-D";
+		status = "ok";
+		reg = <0xc1108d20 0x20>;
+		device_id = <4>;
+		pinctrl-names="default";
+		pinctrl-0=<&d_i2c_master>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+	};
 
 /// ***************************************************************************************
 ///	-	Power
@@ -811,6 +823,14 @@ sdio{
             vddao_voltage = <1200000>;                          /** default voltage for AO, optional     */
             status = "okay";                                    /** status of pmu, only one should be OK */
         };
+		
+		 rtc8563{
+            compatible = "rtc8563";                           /** for driver probe, must have          */
+            i2c_bus = "i2c_bus_ao";                           /** attached i2c bus, must have          */
+            reg = <0x51>;                                     /** device i2c address, must have        */
+            gpio_interrupt = "GPIOX_12";
+            status = "ok";
+        };
     };
 
 /// ***************************************************************************************
@@ -1068,7 +1088,7 @@ sdio{
     audio_m8_audio{
         compatible = "amlogic,m8_audio_codec";
         dev_name = "aml_m8_codec.0";
-        status = "disable";
+        status = "okay";
     };
 
 //$$ DEVICE="dummy_codec"
@@ -1081,10 +1101,10 @@ sdio{
 
     aml_audio_codec{
         compatible = "amlogic,audio_codec";
-          dummy:dummy{
-            codec_name = "dummy_codec";
-            status = "okay";
-          };
+//		dummy:dummy{
+//			codec_name = "dummy_codec";
+//			status = "okay";
+//			};
 //        rt5616:rt5616{
 //            codec_name = "rt5616";
 //            i2c_addr = <0x1B>;
@@ -1110,6 +1130,15 @@ sdio{
 //            capless = <1>;
 //	    status = "disable";
 //        };
+		es8323:es8323{
+           codec_name = "es8323";
+           i2c_addr = <0x10>;
+           i2c_bus = "i2c_bus_d";
+           id_reg = <0x00>;
+           id_val = <0x06>;
+           capless = <0>;
+           status = "okay";
+        };
     };
 
 //$$ DEVICE="aml_m8_sound_card"
@@ -1124,7 +1153,7 @@ sdio{
     aml_m8_sound_card{
         compatible = "sound_card, aml_snd_m8";
         aml,sound_card = "AML-M8AUDIO";
-        aml,codec_dai = "AML-M8","dummy_codec";
+        aml,codec_dai = "AML-M8","es8323-hifi";
         //aml,codec_name = "aml_m8_codec.0","rt5616.4-001b";
         //aml,audio-codec = <&rt5616>;
         aml,audio-routing-rt5616 = 
@@ -1144,6 +1173,11 @@ sdio{
         aml,audio-routing-dummy=
             "Ext Spk","LOUTL",
             "Ext Spk","LOUTR";
+		aml,audio-routing-es8323 = 
+            "Ext Spk","LOUTL",
+            "Ext Spk","LOUTR",
+            "HP","HPOL",
+            "HP","HPOR";
 
         mute_gpio = "GPIO_BSD_EN";
         hp_disable;
@@ -1723,10 +1757,10 @@ sdio{
 //$$ MATCH "I2C_D_pin_match" = "&d_i2c_master"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
 //$$ L2 PROP_STR 2 = "amlogic,pins"
-//		d_i2c_master:d_i2c{
-//			amlogic,setmask=<4 0xc>;
-//			amlogic,pins="GPIOH_7","GPIOH_8";
-//		};
+		d_i2c_master:d_i2c{
+			amlogic,setmask=<4 0xc>;
+			amlogic,pins="GPIOH_7","GPIOH_8";
+		};
 
 //$$ MATCH "remote_pin_0_match" = "&remote_pins"
 //$$ L2 PROP_U32 2 = "amlogic,setmask"
@@ -1741,11 +1775,13 @@ sdio{
 //$$ L2 PROP_U32 4 = "amlogic,clrmask"
 //$$ L2 PROP_STR 8 = "amlogic,pins"
         audio_pins:audio_pin{
-            amlogic,setmask=<3 0x78000000      /*bt pcm*/
+            amlogic,setmask=<
+	    		1 0x20
+	    		3 0x78000000      /*bt pcm*/
                              10 0x78000000>;  /*external audio codec*/
             amlogic,clrmask=<4 0x3c000         /*bt pcm*/            
                              5 0x1000>;        /*bt pcm*/
-            amlogic,pins = "GPIOX_4","GPIOX_5","GPIOX_6","GPIOX_7",        /*bt pcm*/
+            amlogic,pins = "GPIOY_5","GPIOX_4","GPIOX_5","GPIOX_6","GPIOX_7",        /*bt pcm*/
                            "GPIOAO_8","GPIOAO_9","GPIOAO_10","GPIOAO_11";  /*external audio codec*/
         };
 
