From ba75ff9947d4a4b54d9d3191c6ea8d4a8523d5fc Mon Sep 17 00:00:00 2001
From: "Chris \"Koying\" Browet" <cbro@semperpax.com>
Date: Sun, 27 Mar 2016 16:36:42 +0200
Subject: [PATCH] FIX: speedup thumb loading

---
 xbmc/video/VideoThumbLoader.cpp | 26 +++++++++++++++++++++++---
 1 file changed, 23 insertions(+), 3 deletions(-)

diff --git a/xbmc/video/VideoThumbLoader.cpp b/xbmc/video/VideoThumbLoader.cpp
index 7628997..5a44ea7 100644
--- a/xbmc/video/VideoThumbLoader.cpp
+++ b/xbmc/video/VideoThumbLoader.cpp
@@ -27,6 +27,8 @@
 #include "FileItem.h"
 #include "filesystem/DirectoryCache.h"
 #include "filesystem/StackDirectory.h"
+#include "filesystem/File.h"
+#include "filesystem/CurlFile.h"
 #include "guilib/GUIWindowManager.h"
 #include "guilib/StereoscopicsManager.h"
 #include "GUIUserMessages.h"
@@ -364,10 +366,28 @@ bool CVideoThumbLoader::LoadItemLookup(CFileItem* pItem)
   // We can only extract flags/thumbs for file-like items
   if (!pItem->m_bIsFolder && pItem->IsVideo())
   {
-    // An auto-generated thumb may have been cached on a different device - check we have it here
     std::string url = pItem->GetArt("thumb");
-    if (StringUtils::StartsWith(url, "image://video@") && !CTextureCache::GetInstance().HasCachedImage(url))
-      pItem->SetArt("thumb", "");
+    if (!CTextureCache::GetInstance().HasCachedImage(url))
+    {
+      // An auto-generated thumb may have been cached on a different device - check we have it here
+      if (StringUtils::StartsWith(url, "image://video@"))
+        pItem->SetArt("thumb", "");
+
+      // Check invalid redirections
+      else if (URIUtils::IsInternetStream(url))
+      {
+        std::string mimetype;
+        bool valid = XFILE::CCurlFile::GetMimeType(CURL(url), mimetype);
+        if (valid && !StringUtils::StartsWith(mimetype, "image/"))
+          valid = false;
+        if (!valid)
+          pItem->SetArt("thumb", "");
+      }
+
+      // Check if thumb source still exists
+      else if (!XFILE::CFile::Exists(url))
+        pItem->SetArt("thumb", "");
+    }
 
     if (!pItem->HasArt("thumb"))
     {
