diff -Nur a/rtl8189ES/Makefile b/rtl8189ES/Makefile
--- a/rtl8189ES/Makefile	2015-07-21 08:02:53.000000000 +0200
+++ b/rtl8189ES/Makefile	2016-02-06 15:40:22.142681525 +0100
@@ -47,7 +47,7 @@
 CONFIG_RTL8192CU_REDEFINE_1X1 = n
 CONFIG_INTEL_WIDI = n
 CONFIG_WAPI_SUPPORT = n
-CONFIG_EFUSE_CONFIG_FILE = y
+CONFIG_EFUSE_CONFIG_FILE = n
 CONFIG_EXT_CLK = n
 CONFIG_TRAFFIC_PROTECT = y
 CONFIG_LOAD_PHY_PARA_FROM_FILE = y
@@ -58,7 +58,7 @@
 CONFIG_SKIP_SIGNAL_SCALE_MAPPING = n
 CONFIG_80211W = n
 CONFIG_REDUCE_TX_CPU_LOADING = n
-CONFIG_BR_EXT = y
+CONFIG_BR_EXT = n
 CONFIG_ANTENNA_DIVERSITY = n
 ######################## Wake On Lan ##########################
 CONFIG_WOWLAN = n
@@ -105,7 +105,7 @@
 CONFIG_PLATFORM_ARM_SUNxI = n
 CONFIG_PLATFORM_ARM_SUN6I = n
 CONFIG_PLATFORM_ARM_SUN7I = n
-CONFIG_PLATFORM_ARM_SUN8I = n
+CONFIG_PLATFORM_ARM_SUN8I = y
 CONFIG_PLATFORM_ACTIONS_ATM702X = n
 CONFIG_PLATFORM_ACTIONS_ATM705X = n
 CONFIG_PLATFORM_ACTIONS_ATV5201 = n
@@ -114,7 +114,7 @@
 CONFIG_PLATFORM_ARM_SPREADTRUM_8810 = n
 CONFIG_PLATFORM_ARM_WMT = n
 CONFIG_PLATFORM_TI_DM365 = n
-CONFIG_PLATFORM_AML = y
+CONFIG_PLATFORM_AML = n
 CONFIG_PLATFORM_MOZART = n
 CONFIG_PLATFORM_RTK119X = n
 ###############################################################
@@ -1303,7 +1303,7 @@
 EXTRA_CFLAGS += -DCONFIG_PLATFORM_ARM_SUN8I
 EXTRA_CFLAGS += -DCONFIG_TRAFFIC_PROTECT
 # default setting for Android 4.1, 4.2
-EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
+# EXTRA_CFLAGS += -DCONFIG_CONCURRENT_MODE
 EXTRA_CFLAGS += -DCONFIG_IOCTL_CFG80211 -DRTW_USE_CFG80211_STA_EVENT
 EXTRA_CFLAGS += -DCONFIG_P2P_IPS -DCONFIG_QOS_OPTIMIZATION
 
diff -Nur a/rtl8189ES/os_dep/linux/sdio_intf.c b/rtl8189ES/os_dep/linux/sdio_intf.c
--- a/rtl8189ES/os_dep/linux/sdio_intf.c	2015-07-21 08:02:53.000000000 +0200
+++ b/rtl8189ES/os_dep/linux/sdio_intf.c	2016-02-06 16:03:39.616199988 +0100
@@ -34,8 +34,6 @@
 #endif
 static int wlan_en_gpio = -1;
 #endif //CONFIG_PLATFORM_INTEL_BYT
-extern void wifi_teardown_dt(void);
-extern int wifi_setup_dt(void);
 
 #ifndef dev_to_sdio_func
 #define dev_to_sdio_func(d)     container_of(d, struct sdio_func, dev)
@@ -967,19 +965,12 @@
 	DBG_871X_LEVEL(_drv_always_, DRV_NAME" BT-Coex version = %s\n", BTCOEXVERSION);
 #endif // BTCOEXVERSION
 
-	ret =wifi_setup_dt();
-	if(ret)
-	{
-		DBG_871X("%s: setup dt failed!!(%d)\n", __FUNCTION__, ret);
-		ret = -1;
-		goto exit;
-	}
 	ret = platform_wifi_power_on();
 	if (ret)
 	{
 		DBG_871X("%s: power on failed!!(%d)\n", __FUNCTION__, ret);
 		ret = -1;
-		goto resource;
+		goto exit;
 	}
 
 	sdio_drvpriv.drv_registered = _TRUE;
@@ -1005,8 +996,6 @@
 
 poweroff:
 	platform_wifi_power_off();
-resource:
-	wifi_teardown_dt();
 
 exit:
 	DBG_871X_LEVEL(_drv_always_, "module init ret=%d\n", ret);
@@ -1025,7 +1014,6 @@
 
 	platform_wifi_power_off();
 
-	wifi_teardown_dt();
 	rtw_suspend_lock_uninit();
 	rtw_drv_proc_deinit();
 	rtw_ndev_notifier_unregister();
diff -Nur a/rtl8189ES/platform/platform_ARM_SUNnI_sdio.c b/rtl8189ES/platform/platform_ARM_SUNnI_sdio.c
--- a/rtl8189ES/platform/platform_ARM_SUNnI_sdio.c	2015-07-21 08:02:53.000000000 +0200
+++ b/rtl8189ES/platform/platform_ARM_SUNnI_sdio.c	2016-02-06 15:42:10.629081751 +0100
@@ -70,6 +70,8 @@
 		sdc_id = val.val;
 		DBG_871X("----- %s sdc_id: %d, mod_sel: %d\n", __FUNCTION__, sdc_id, mod_sel);
 
+		wifi_pm_power(0);
+		mdelay(50);
 #if defined(CONFIG_PLATFORM_ARM_SUN6I) || defined(CONFIG_PLATFORM_ARM_SUN7I)
 		sw_mci_rescan_card(sdc_id, 1);
 #elif defined(CONFIG_PLATFORM_ARM_SUN8I)
@@ -82,12 +84,7 @@
 	}
 
 #ifdef CONFIG_GPIO_WAKEUP
-#ifdef CONFIG_RTL8723B
-	type = script_get_item("wifi_para", "rtl8723bs_wl_host_wake", &val);
-#endif
-#ifdef CONFIG_RTL8188E
-	type = script_get_item("wifi_para", "rtl8189es_host_wake", &val);
-#endif
+	type = script_get_item("wifi_para", "wl_host_wake", &val);
 	if (SCIRPT_ITEM_VALUE_TYPE_PIO != type) {
 		DBG_871X("No definition of wake up host PIN\n");
 		ret = -1;
